{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "name": "Genomics_Workshop.ipynb",
      "provenance": [],
      "machine_shape": "hm",
      "gpuType": "T4",
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.8.10"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Aksinhaa/ColabFold/blob/main/Genomics_Workshop.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "G4yBrceuFbf3"
      },
      "source": [
        "<video src=\"https://raw.githubusercontent.com/PoODL-CES/PoODL_NGS_workshop.io/main/Nature%20of%20Science.mp4\" controls autoplay loop width=\"400\" align=\"right\"></video>\n",
        "\n",
        "## Genomics Learning Workshop: NGS Analysis Pipeline\n",
        "\n",
        "Easy-to-follow workflow for processing Illumina whole-genome resequencing reads for population genomics. Steps include trimming, mapping, sorting, variant calling, variant filtering, PCA, and ADMIXTURE analysis. For more details, see the bottom of this notebook and visit the GitHub repository.\n",
        "Repository link: [Genomics Learning Workshop](https://github.com/PoODL-CES/Genomics_learning_workshop)\n",
        "\n",
        "This workshop introduces:\n",
        "- Handling raw FASTQ files\n",
        "- Performing quality control and trimming\n",
        "- Mapping reads to a reference genome\n",
        "- Calling and filtering variants\n",
        "- Running PCA and ADMIXTURE for population analysis\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "ee74ca19",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f52e5784-7290-477c-9bf1-274cad7d73bd"
      },
      "source": [
        "#@title Install system-level NGS bioinformatics tools\n",
        "%%time\n",
        "\n",
        "# Install samtools, fastqc, vcftools using apt-get (stable)\n",
        "!apt-get update -qq\n",
        "!apt-get install -y samtools fastqc vcftools\n",
        "\n",
        "# Install cutadapt (requirement for Trim Galore, also installed via conda later)\n",
        "!pip install --quiet cutadapt"
      ],
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  at-spi2-core default-jre default-jre-headless fonts-dejavu-core\n",
            "  fonts-dejavu-extra gsettings-desktop-schemas libapache-pom-java\n",
            "  libargs4j-java libatk-bridge2.0-0 libatk-wrapper-java\n",
            "  libatk-wrapper-java-jni libatk1.0-0 libatk1.0-data libatspi2.0-0\n",
            "  libcommons-compress-java libcommons-io-java libcommons-jexl2-java\n",
            "  libcommons-lang3-java libcommons-logging-java libcommons-math3-java\n",
            "  libcommons-parent-java libfindbin-libs-perl libhts3 libhtscodecs2\n",
            "  libhtsjdk-java libjbzip2-java libjson-simple-java libngs-java libngs-sdk-dev\n",
            "  libngs-sdk2 libsis-base-java libsis-base-jni libsis-jhdf5-java\n",
            "  libsis-jhdf5-jni libsnappy-java libsnappy-jni libxcomposite1 libxtst6\n",
            "  libxxf86dga1 openjdk-11-jre openjdk-11-jre-headless session-migration tabix\n",
            "  x11-utils\n",
            "Suggested packages:\n",
            "  libxz-java libcommons-io-java-doc libcommons-lang3-java-doc\n",
            "  libavalon-framework-java libcommons-logging-java-doc\n",
            "  libexcalibur-logkit-java liblog4j1.2-java picard-tools libjson-simple-doc\n",
            "  libnss-mdns fonts-ipafont-gothic fonts-ipafont-mincho fonts-wqy-microhei\n",
            "  | fonts-wqy-zenhei fonts-indic cwltool mesa-utils\n",
            "The following NEW packages will be installed:\n",
            "  at-spi2-core default-jre default-jre-headless fastqc fonts-dejavu-core\n",
            "  fonts-dejavu-extra gsettings-desktop-schemas libapache-pom-java\n",
            "  libargs4j-java libatk-bridge2.0-0 libatk-wrapper-java\n",
            "  libatk-wrapper-java-jni libatk1.0-0 libatk1.0-data libatspi2.0-0\n",
            "  libcommons-compress-java libcommons-io-java libcommons-jexl2-java\n",
            "  libcommons-lang3-java libcommons-logging-java libcommons-math3-java\n",
            "  libcommons-parent-java libfindbin-libs-perl libhts3 libhtscodecs2\n",
            "  libhtsjdk-java libjbzip2-java libjson-simple-java libngs-java libngs-sdk-dev\n",
            "  libngs-sdk2 libsis-base-java libsis-base-jni libsis-jhdf5-java\n",
            "  libsis-jhdf5-jni libsnappy-java libsnappy-jni libxcomposite1 libxtst6\n",
            "  libxxf86dga1 openjdk-11-jre openjdk-11-jre-headless samtools\n",
            "  session-migration tabix vcftools x11-utils\n",
            "0 upgraded, 47 newly installed, 0 to remove and 50 not upgraded.\n",
            "Need to get 56.3 MB of archives.\n",
            "After this operation, 207 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatspi2.0-0 amd64 2.44.0-3 [80.9 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/main amd64 session-migration amd64 0.3.6 [9,774 B]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 gsettings-desktop-schemas all 42.0-1ubuntu1 [31.1 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/main amd64 at-spi2-core amd64 2.44.0-3 [54.4 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre-headless amd64 11.0.29+7-1ubuntu1~22.04 [42.6 MB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre-headless amd64 2:1.11-72build2 [3,042 B]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 openjdk-11-jre amd64 11.0.29+7-1ubuntu1~22.04 [214 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/main amd64 default-jre amd64 2:1.11-72build2 [896 B]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-math3-java all 3.6.1-3 [2,016 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjbzip2-java all 0.9.1-7 [42.8 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libapache-pom-java all 18-1 [4,720 B]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-parent-java all 43-1 [10.8 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-compress-java all 1.21-1 [575 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-logging-java all 1.2-2 [60.3 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-jexl2-java all 2.1.1-5 [251 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libjson-simple-java all 2.3.0-1 [61.0 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk2 amd64 2.11.2-3 [65.6 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-sdk-dev amd64 2.11.2-3 [111 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libngs-java amd64 2.11.2-3 [75.8 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-jni amd64 1.1.8.3-1 [6,756 B]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsnappy-java all 1.1.8.3-1 [93.9 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtsjdk-java all 2.24.1+dfsg-1 [1,644 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libfindbin-libs-perl all 2.190.02-1 [23.2 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libargs4j-java all 2.33-1.1 [137 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-io-java all 2.11.0-2 [297 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libcommons-lang3-java all 3.11-1 [526 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-jni amd64 18.09~pre1+git20180928.45fbd31+dfsg-3 [12.7 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-base-java all 18.09~pre1+git20180928.45fbd31+dfsg-3 [96.4 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-jni amd64 19.04.0+dfsg-4 [96.6 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libsis-jhdf5-java all 19.04.0+dfsg-4 [1,490 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu jammy/universe amd64 fastqc all 0.11.9+dfsg-5 [370 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk1.0-data all 2.36.0-3build1 [2,824 B]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk1.0-0 amd64 2.36.0-3build1 [51.9 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-bridge2.0-0 amd64 2.38.0-3 [66.6 kB]\n",
            "Get:38 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcomposite1 amd64 1:0.4.5-1build2 [7,192 B]\n",
            "Get:39 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n",
            "Get:40 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n",
            "Get:41 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n",
            "Get:42 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n",
            "Get:43 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
            "Get:44 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
            "Get:45 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
            "Get:46 http://archive.ubuntu.com/ubuntu jammy/universe amd64 tabix amd64 1.13+ds-2build1 [351 kB]\n",
            "Get:47 http://archive.ubuntu.com/ubuntu jammy/universe amd64 vcftools amd64 0.1.16-3 [393 kB]\n",
            "Fetched 56.3 MB in 1s (57.6 MB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package libatspi2.0-0:amd64.\n",
            "(Reading database ... 121713 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libatspi2.0-0_2.44.0-3_amd64.deb ...\n",
            "Unpacking libatspi2.0-0:amd64 (2.44.0-3) ...\n",
            "Selecting previously unselected package libxtst6:amd64.\n",
            "Preparing to unpack .../01-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n",
            "Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Selecting previously unselected package session-migration.\n",
            "Preparing to unpack .../02-session-migration_0.3.6_amd64.deb ...\n",
            "Unpacking session-migration (0.3.6) ...\n",
            "Selecting previously unselected package gsettings-desktop-schemas.\n",
            "Preparing to unpack .../03-gsettings-desktop-schemas_42.0-1ubuntu1_all.deb ...\n",
            "Unpacking gsettings-desktop-schemas (42.0-1ubuntu1) ...\n",
            "Selecting previously unselected package at-spi2-core.\n",
            "Preparing to unpack .../04-at-spi2-core_2.44.0-3_amd64.deb ...\n",
            "Unpacking at-spi2-core (2.44.0-3) ...\n",
            "Selecting previously unselected package openjdk-11-jre-headless:amd64.\n",
            "Preparing to unpack .../05-openjdk-11-jre-headless_11.0.29+7-1ubuntu1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-11-jre-headless:amd64 (11.0.29+7-1ubuntu1~22.04) ...\n",
            "Selecting previously unselected package default-jre-headless.\n",
            "Preparing to unpack .../06-default-jre-headless_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre-headless (2:1.11-72build2) ...\n",
            "Selecting previously unselected package openjdk-11-jre:amd64.\n",
            "Preparing to unpack .../07-openjdk-11-jre_11.0.29+7-1ubuntu1~22.04_amd64.deb ...\n",
            "Unpacking openjdk-11-jre:amd64 (11.0.29+7-1ubuntu1~22.04) ...\n",
            "Selecting previously unselected package default-jre.\n",
            "Preparing to unpack .../08-default-jre_2%3a1.11-72build2_amd64.deb ...\n",
            "Unpacking default-jre (2:1.11-72build2) ...\n",
            "Selecting previously unselected package libcommons-math3-java.\n",
            "Preparing to unpack .../09-libcommons-math3-java_3.6.1-3_all.deb ...\n",
            "Unpacking libcommons-math3-java (3.6.1-3) ...\n",
            "Selecting previously unselected package libjbzip2-java.\n",
            "Preparing to unpack .../10-libjbzip2-java_0.9.1-7_all.deb ...\n",
            "Unpacking libjbzip2-java (0.9.1-7) ...\n",
            "Selecting previously unselected package libapache-pom-java.\n",
            "Preparing to unpack .../11-libapache-pom-java_18-1_all.deb ...\n",
            "Unpacking libapache-pom-java (18-1) ...\n",
            "Selecting previously unselected package libcommons-parent-java.\n",
            "Preparing to unpack .../12-libcommons-parent-java_43-1_all.deb ...\n",
            "Unpacking libcommons-parent-java (43-1) ...\n",
            "Selecting previously unselected package libcommons-compress-java.\n",
            "Preparing to unpack .../13-libcommons-compress-java_1.21-1_all.deb ...\n",
            "Unpacking libcommons-compress-java (1.21-1) ...\n",
            "Selecting previously unselected package libcommons-logging-java.\n",
            "Preparing to unpack .../14-libcommons-logging-java_1.2-2_all.deb ...\n",
            "Unpacking libcommons-logging-java (1.2-2) ...\n",
            "Selecting previously unselected package libcommons-jexl2-java.\n",
            "Preparing to unpack .../15-libcommons-jexl2-java_2.1.1-5_all.deb ...\n",
            "Unpacking libcommons-jexl2-java (2.1.1-5) ...\n",
            "Selecting previously unselected package libjson-simple-java.\n",
            "Preparing to unpack .../16-libjson-simple-java_2.3.0-1_all.deb ...\n",
            "Unpacking libjson-simple-java (2.3.0-1) ...\n",
            "Selecting previously unselected package libngs-sdk2:amd64.\n",
            "Preparing to unpack .../17-libngs-sdk2_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-sdk-dev:amd64.\n",
            "Preparing to unpack .../18-libngs-sdk-dev_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libngs-java:amd64.\n",
            "Preparing to unpack .../19-libngs-java_2.11.2-3_amd64.deb ...\n",
            "Unpacking libngs-java:amd64 (2.11.2-3) ...\n",
            "Selecting previously unselected package libsnappy-jni.\n",
            "Preparing to unpack .../20-libsnappy-jni_1.1.8.3-1_amd64.deb ...\n",
            "Unpacking libsnappy-jni (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libsnappy-java.\n",
            "Preparing to unpack .../21-libsnappy-java_1.1.8.3-1_all.deb ...\n",
            "Unpacking libsnappy-java (1.1.8.3-1) ...\n",
            "Selecting previously unselected package libhtsjdk-java.\n",
            "Preparing to unpack .../22-libhtsjdk-java_2.24.1+dfsg-1_all.deb ...\n",
            "Unpacking libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Selecting previously unselected package libfindbin-libs-perl.\n",
            "Preparing to unpack .../23-libfindbin-libs-perl_2.190.02-1_all.deb ...\n",
            "Unpacking libfindbin-libs-perl (2.190.02-1) ...\n",
            "Selecting previously unselected package libargs4j-java.\n",
            "Preparing to unpack .../24-libargs4j-java_2.33-1.1_all.deb ...\n",
            "Unpacking libargs4j-java (2.33-1.1) ...\n",
            "Selecting previously unselected package libcommons-io-java.\n",
            "Preparing to unpack .../25-libcommons-io-java_2.11.0-2_all.deb ...\n",
            "Unpacking libcommons-io-java (2.11.0-2) ...\n",
            "Selecting previously unselected package libcommons-lang3-java.\n",
            "Preparing to unpack .../26-libcommons-lang3-java_3.11-1_all.deb ...\n",
            "Unpacking libcommons-lang3-java (3.11-1) ...\n",
            "Selecting previously unselected package libsis-base-jni.\n",
            "Preparing to unpack .../27-libsis-base-jni_18.09~pre1+git20180928.45fbd31+dfsg-3_amd64.deb ...\n",
            "Unpacking libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-base-java.\n",
            "Preparing to unpack .../28-libsis-base-java_18.09~pre1+git20180928.45fbd31+dfsg-3_all.deb ...\n",
            "Unpacking libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Selecting previously unselected package libsis-jhdf5-jni.\n",
            "Preparing to unpack .../29-libsis-jhdf5-jni_19.04.0+dfsg-4_amd64.deb ...\n",
            "Unpacking libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package libsis-jhdf5-java.\n",
            "Preparing to unpack .../30-libsis-jhdf5-java_19.04.0+dfsg-4_all.deb ...\n",
            "Unpacking libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Selecting previously unselected package fastqc.\n",
            "Preparing to unpack .../31-fastqc_0.11.9+dfsg-5_all.deb ...\n",
            "Unpacking fastqc (0.11.9+dfsg-5) ...\n",
            "Selecting previously unselected package fonts-dejavu-core.\n",
            "Preparing to unpack .../32-fonts-dejavu-core_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-core (2.37-2build1) ...\n",
            "Selecting previously unselected package fonts-dejavu-extra.\n",
            "Preparing to unpack .../33-fonts-dejavu-extra_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-extra (2.37-2build1) ...\n",
            "Selecting previously unselected package libatk1.0-data.\n",
            "Preparing to unpack .../34-libatk1.0-data_2.36.0-3build1_all.deb ...\n",
            "Unpacking libatk1.0-data (2.36.0-3build1) ...\n",
            "Selecting previously unselected package libatk1.0-0:amd64.\n",
            "Preparing to unpack .../35-libatk1.0-0_2.36.0-3build1_amd64.deb ...\n",
            "Unpacking libatk1.0-0:amd64 (2.36.0-3build1) ...\n",
            "Selecting previously unselected package libatk-bridge2.0-0:amd64.\n",
            "Preparing to unpack .../36-libatk-bridge2.0-0_2.38.0-3_amd64.deb ...\n",
            "Unpacking libatk-bridge2.0-0:amd64 (2.38.0-3) ...\n",
            "Selecting previously unselected package libxcomposite1:amd64.\n",
            "Preparing to unpack .../37-libxcomposite1_1%3a0.4.5-1build2_amd64.deb ...\n",
            "Unpacking libxcomposite1:amd64 (1:0.4.5-1build2) ...\n",
            "Selecting previously unselected package libxxf86dga1:amd64.\n",
            "Preparing to unpack .../38-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n",
            "Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Selecting previously unselected package x11-utils.\n",
            "Preparing to unpack .../39-x11-utils_7.7+5build2_amd64.deb ...\n",
            "Unpacking x11-utils (7.7+5build2) ...\n",
            "Selecting previously unselected package libatk-wrapper-java.\n",
            "Preparing to unpack .../40-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n",
            "Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n",
            "Preparing to unpack .../41-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n",
            "Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libhtscodecs2:amd64.\n",
            "Preparing to unpack .../42-libhtscodecs2_1.1.1-3_amd64.deb ...\n",
            "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../43-libhts3_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../44-samtools_1.13-4_amd64.deb ...\n",
            "Unpacking samtools (1.13-4) ...\n",
            "Selecting previously unselected package tabix.\n",
            "Preparing to unpack .../45-tabix_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking tabix (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package vcftools.\n",
            "Preparing to unpack .../46-vcftools_0.1.16-3_amd64.deb ...\n",
            "Unpacking vcftools (0.1.16-3) ...\n",
            "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Setting up vcftools (0.1.16-3) ...\n",
            "Setting up session-migration (0.3.6) ...\n",
            "Created symlink /etc/systemd/user/graphical-session-pre.target.wants/session-migration.service → /usr/lib/systemd/user/session-migration.service.\n",
            "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Setting up openjdk-11-jre-headless:amd64 (11.0.29+7-1ubuntu1~22.04) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/jjs to provide /usr/bin/jjs (jjs) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/rmid to provide /usr/bin/rmid (rmid) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/pack200 to provide /usr/bin/pack200 (pack200) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-11-openjdk-amd64/bin/unpack200 to provide /usr/bin/unpack200 (unpack200) in auto mode\n",
            "Setting up libngs-sdk2:amd64 (2.11.2-3) ...\n",
            "Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Setting up openjdk-11-jre:amd64 (11.0.29+7-1ubuntu1~22.04) ...\n",
            "Setting up libfindbin-libs-perl (2.190.02-1) ...\n",
            "Setting up libatspi2.0-0:amd64 (2.44.0-3) ...\n",
            "Setting up libjson-simple-java (2.3.0-1) ...\n",
            "Setting up samtools (1.13-4) ...\n",
            "Setting up libsis-base-jni (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libapache-pom-java (18-1) ...\n",
            "Setting up fonts-dejavu-core (2.37-2build1) ...\n",
            "Setting up libjbzip2-java (0.9.1-7) ...\n",
            "Setting up libngs-sdk-dev:amd64 (2.11.2-3) ...\n",
            "Setting up libatk1.0-data (2.36.0-3build1) ...\n",
            "Setting up fonts-dejavu-extra (2.37-2build1) ...\n",
            "Setting up libcommons-math3-java (3.6.1-3) ...\n",
            "Setting up libsis-jhdf5-jni (19.04.0+dfsg-4) ...\n",
            "Setting up libatk1.0-0:amd64 (2.36.0-3build1) ...\n",
            "Setting up libngs-java:amd64 (2.11.2-3) ...\n",
            "Setting up tabix (1.13+ds-2build1) ...\n",
            "Setting up libsnappy-jni (1.1.8.3-1) ...\n",
            "Setting up libxcomposite1:amd64 (1:0.4.5-1build2) ...\n",
            "Setting up libargs4j-java (2.33-1.1) ...\n",
            "Setting up gsettings-desktop-schemas (42.0-1ubuntu1) ...\n",
            "Setting up default-jre-headless (2:1.11-72build2) ...\n",
            "Setting up libcommons-parent-java (43-1) ...\n",
            "Setting up libcommons-logging-java (1.2-2) ...\n",
            "Setting up default-jre (2:1.11-72build2) ...\n",
            "Setting up libatk-bridge2.0-0:amd64 (2.38.0-3) ...\n",
            "Setting up libcommons-jexl2-java (2.1.1-5) ...\n",
            "Setting up libcommons-lang3-java (3.11-1) ...\n",
            "Setting up x11-utils (7.7+5build2) ...\n",
            "Setting up libsnappy-java (1.1.8.3-1) ...\n",
            "Setting up libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Setting up libcommons-compress-java (1.21-1) ...\n",
            "Setting up libcommons-io-java (2.11.0-2) ...\n",
            "Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Setting up libsis-base-java (18.09~pre1+git20180928.45fbd31+dfsg-3) ...\n",
            "Setting up libhtsjdk-java (2.24.1+dfsg-1) ...\n",
            "Setting up libsis-jhdf5-java (19.04.0+dfsg-4) ...\n",
            "Setting up fastqc (0.11.9+dfsg-5) ...\n",
            "Processing triggers for mailcap (3.70+nmu1ubuntu1) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for libglib2.0-0:amd64 (2.72.4-0ubuntu2.6) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Setting up at-spi2-core (2.44.0-3) ...\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m281.1/281.1 kB\u001b[0m \u001b[31m11.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m106.1/106.1 kB\u001b[0m \u001b[31m10.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m264.7/264.7 kB\u001b[0m \u001b[31m24.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m132.4/132.4 kB\u001b[0m \u001b[31m10.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hCPU times: user 1.38 s, sys: 211 ms, total: 1.59 s\n",
            "Wall time: 29.3 s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "f6704e35",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "de093573-ff74-4b48-d436-59bd81b9eceb"
      },
      "source": [
        "# Miniconda installation and environment setup for Colab NGS Workshop\n",
        "\n",
        "# Download and install Miniconda (skip if already installed)\n",
        "!wget -q https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh\n",
        "!bash miniconda.sh -b -p /usr/local/miniconda\n",
        "\n",
        "import sys, os\n",
        "sys.path.append('/usr/local/miniconda/lib/python3.8/site-packages')\n",
        "os.environ['PATH'] = \"/usr/local/miniconda/bin:\" + os.environ['PATH']\n",
        "\n",
        "# Explicitly clear potentially problematic environment variables from Python's os.environ\n",
        "if 'CONDA_PREFIX' in os.environ:\n",
        "    del os.environ['CONDA_PREFIX']\n",
        "if 'CONDA_ENVS_PATH' in os.environ:\n",
        "    del os.environ['CONDA_ENVS_PATH']\n",
        "\n",
        "# Accept ToS for main and R conda channels\n",
        "!conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/main\n",
        "!conda tos accept --override-channels --channel https://repo.anaconda.com/pkgs/r\n",
        "\n",
        "# Create new conda environment for your workshop if it doesn't exist\n",
        "!if ! conda info --envs | grep -w 'workshop_ngs'; then \\\n",
        "    conda create -y -n workshop_ngs python=3.7; \\\n",
        "else \\\n",
        "    echo \"Environment 'workshop_ngs' already exists, skipping creation.\"; \\\n",
        "fi\n",
        "\n",
        "# Install necessary bioinformatics tools into the environment\n",
        "!conda install -y -n workshop_ngs -c bioconda -c conda-forge trim-galore samtools fastqc vcftools gatk4\n",
        "\n",
        "# Verify tool versions inside activated environment in one shell session\n",
        "!bash -c \"source /usr/local/miniconda/bin/activate workshop_ngs && \\\n",
        "          trim_galore --version && samtools --version && fastqc --version && \\\n",
        "          vcftools --version && gatk --version\""
      ],
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "PREFIX=/usr/local/miniconda\n",
            "Unpacking bootstrapper...\n",
            "Unpacking payload...\n",
            "\n",
            "Installing base environment...\n",
            "\n",
            "Preparing transaction: ...working... done\n",
            "Executing transaction: ...working... done\n",
            "installation finished.\n",
            "WARNING:\n",
            "    You currently have a PYTHONPATH environment variable set. This may cause\n",
            "    unexpected behavior when running the Python interpreter in Miniconda3.\n",
            "    For best results, please verify that your PYTHONPATH only points to\n",
            "    directories of packages that are compatible with the Python interpreter\n",
            "    in Miniconda3: /usr/local/miniconda\n",
            "accepted Terms of Service for \u001b[4;94mhttps://repo.anaconda.com/pkgs/main\u001b[0m\n",
            "accepted Terms of Service for \u001b[4;94mhttps://repo.anaconda.com/pkgs/r\u001b[0m\n",
            "\u001b[1;33mJupyter detected\u001b[0m\u001b[1;33m...\u001b[0m\n",
            "\u001b[1;32m2\u001b[0m\u001b[1;32m channel Terms of Service accepted\u001b[0m\n",
            "Retrieving notices: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Channels:\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local/miniconda/envs/workshop_ngs\n",
            "\n",
            "  added / updated specs:\n",
            "    - python=3.7\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    certifi-2022.12.7          |   py37h06a4308_0         150 KB\n",
            "    openssl-1.1.1w             |       h7f8727e_0         3.7 MB\n",
            "    pip-22.3.1                 |   py37h06a4308_0         2.7 MB\n",
            "    python-3.7.16              |       h7a1cb2a_0        44.8 MB\n",
            "    setuptools-65.6.3          |   py37h06a4308_0         1.1 MB\n",
            "    wheel-0.38.4               |   py37h06a4308_0          63 KB\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        52.6 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main \n",
            "  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-5.1-1_gnu \n",
            "  ca-certificates    pkgs/main/linux-64::ca-certificates-2025.11.4-h06a4308_0 \n",
            "  certifi            pkgs/main/linux-64::certifi-2022.12.7-py37h06a4308_0 \n",
            "  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.44-h153f514_2 \n",
            "  libffi             pkgs/main/linux-64::libffi-3.4.4-h6a678d5_1 \n",
            "  libgcc             pkgs/main/linux-64::libgcc-15.2.0-h69a1729_7 \n",
            "  libgcc-ng          pkgs/main/linux-64::libgcc-ng-15.2.0-h166f726_7 \n",
            "  libgomp            pkgs/main/linux-64::libgomp-15.2.0-h4751f2c_7 \n",
            "  libstdcxx          pkgs/main/linux-64::libstdcxx-15.2.0-h39759b7_7 \n",
            "  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-15.2.0-hc03a8fd_7 \n",
            "  libxcb             pkgs/main/linux-64::libxcb-1.17.0-h9b100fa_0 \n",
            "  libzlib            pkgs/main/linux-64::libzlib-1.3.1-hb25bd0a_0 \n",
            "  ncurses            pkgs/main/linux-64::ncurses-6.5-h7934f7d_0 \n",
            "  openssl            pkgs/main/linux-64::openssl-1.1.1w-h7f8727e_0 \n",
            "  pip                pkgs/main/linux-64::pip-22.3.1-py37h06a4308_0 \n",
            "  pthread-stubs      pkgs/main/linux-64::pthread-stubs-0.3-h0ce48e5_1 \n",
            "  python             pkgs/main/linux-64::python-3.7.16-h7a1cb2a_0 \n",
            "  readline           pkgs/main/linux-64::readline-8.3-hc2a1206_0 \n",
            "  setuptools         pkgs/main/linux-64::setuptools-65.6.3-py37h06a4308_0 \n",
            "  sqlite             pkgs/main/linux-64::sqlite-3.51.0-h2a70700_0 \n",
            "  tk                 pkgs/main/linux-64::tk-8.6.15-h54e0aa7_0 \n",
            "  wheel              pkgs/main/linux-64::wheel-0.38.4-py37h06a4308_0 \n",
            "  xorg-libx11        pkgs/main/linux-64::xorg-libx11-1.8.12-h9b100fa_1 \n",
            "  xorg-libxau        pkgs/main/linux-64::xorg-libxau-1.0.12-h9b100fa_0 \n",
            "  xorg-libxdmcp      pkgs/main/linux-64::xorg-libxdmcp-1.1.5-h9b100fa_0 \n",
            "  xorg-xorgproto     pkgs/main/linux-64::xorg-xorgproto-2024.1-h5eee18b_1 \n",
            "  xz                 pkgs/main/linux-64::xz-5.6.4-h5eee18b_1 \n",
            "  zlib               pkgs/main/linux-64::zlib-1.3.1-hb25bd0a_0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "python-3.7.16        | 44.8 MB   | :   0% 0/1 [00:00<?, ?it/s]\n",
            "openssl-1.1.1w       | 3.7 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "pip-22.3.1           | 2.7 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "setuptools-65.6.3    | 1.1 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "certifi-2022.12.7    | 150 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "python-3.7.16        | 44.8 MB   | :   0% 0.00034843833210413685/1 [00:00<14:37, 878.19s/it]\n",
            "\n",
            "\n",
            "\n",
            "certifi-2022.12.7    | 150 KB    | :  11% 0.10679390158848107/1 [00:00<00:02,  2.82s/it]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openssl-1.1.1w       | 3.7 MB    | :   0% 0.004187193943698291/1 [00:00<01:15, 75.88s/it]\u001b[A\n",
            "\n",
            "pip-22.3.1           | 2.7 MB    | :   1% 0.00579954747671544/1 [00:00<00:55, 55.45s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "certifi-2022.12.7    | 150 KB    | : 100% 1.0/1 [00:00<00:00,  2.82s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "setuptools-65.6.3    | 1.1 MB    | :   1% 0.01372742712122012/1 [00:00<00:23, 23.75s/it]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "wheel-0.38.4         | 63 KB     | :  25% 0.25267963171450164/1 [00:00<00:01,  1.48s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "wheel-0.38.4         | 63 KB     | : 100% 1.0/1 [00:00<00:00,  1.48s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "python-3.7.16        | 44.8 MB   | :   9% 0.08745802135813835/1 [00:00<00:03,  3.64s/it]    \n",
            "\n",
            "\n",
            "setuptools-65.6.3    | 1.1 MB    | : 100% 1.0/1 [00:00<00:00, 23.75s/it]                \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "pip-22.3.1           | 2.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.64it/s]                \u001b[A\u001b[A\n",
            "\n",
            "pip-22.3.1           | 2.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.64it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "wheel-0.38.4         | 63 KB     | : 100% 1.0/1 [00:00<00:00,  2.45it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "python-3.7.16        | 44.8 MB   | :  20% 0.2031395476167118/1 [00:00<00:01,  1.83s/it] \n",
            "openssl-1.1.1w       | 3.7 MB    | : 100% 1.0/1 [00:00<00:00,  2.33it/s]                 \u001b[A\n",
            "python-3.7.16        | 44.8 MB   | :  90% 0.9000162118249855/1 [00:00<00:00,  1.47it/s]\n",
            "\n",
            "\n",
            "setuptools-65.6.3    | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.13it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "setuptools-65.6.3    | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.13it/s]\u001b[A\u001b[A\u001b[A\n",
            "openssl-1.1.1w       | 3.7 MB    | : 100% 1.0/1 [00:01<00:00,  2.33it/s]\u001b[A\n",
            "\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: - \b\b\\ \b\b| \b\bdone\n",
            "Verifying transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Executing transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "#\n",
            "# To activate this environment, use\n",
            "#\n",
            "#     $ conda activate workshop_ngs\n",
            "#\n",
            "# To deactivate an active environment, use\n",
            "#\n",
            "#     $ conda deactivate\n",
            "\n",
            "\u001b[1;33mJupyter detected\u001b[0m\u001b[1;33m...\u001b[0m\n",
            "\u001b[1;32m2\u001b[0m\u001b[1;32m channel Terms of Service accepted\u001b[0m\n",
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            " - defaults\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local/miniconda/envs/workshop_ngs\n",
            "\n",
            "  added / updated specs:\n",
            "    - fastqc\n",
            "    - gatk4\n",
            "    - samtools\n",
            "    - trim-galore\n",
            "    - vcftools\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    alsa-lib-1.2.14            |       hb9d3cd8_0         553 KB  conda-forge\n",
            "    bzip2-1.0.8                |       hda65f42_8         254 KB  conda-forge\n",
            "    c-ares-1.34.5              |       hb9d3cd8_0         202 KB  conda-forge\n",
            "    ca-certificates-2025.11.12 |       hbd8a1cb_0         149 KB  conda-forge\n",
            "    cairo-1.18.4               |       h3394656_0         955 KB  conda-forge\n",
            "    certifi-2024.8.30          |     pyhd8ed1ab_0         160 KB  conda-forge\n",
            "    cutadapt-4.4               |   py37h8902056_0         263 KB  bioconda\n",
            "    dnaio-0.10.0               |   py37h8902056_1          90 KB  bioconda\n",
            "    fastqc-0.12.1              |       hdfd78af_0        11.1 MB  bioconda\n",
            "    font-ttf-dejavu-sans-mono-2.37|       hab24e00_0         388 KB  conda-forge\n",
            "    font-ttf-inconsolata-3.000 |       h77eed37_0          94 KB  conda-forge\n",
            "    font-ttf-source-code-pro-2.038|       h77eed37_0         684 KB  conda-forge\n",
            "    font-ttf-ubuntu-0.83       |       h77eed37_3         1.5 MB  conda-forge\n",
            "    fontconfig-2.15.0          |       h7e30c49_1         259 KB  conda-forge\n",
            "    fonts-conda-ecosystem-1    |                0           4 KB  conda-forge\n",
            "    fonts-conda-forge-1        |       hc364b38_1           4 KB  conda-forge\n",
            "    freetype-2.14.1            |       ha770c72_0         169 KB  conda-forge\n",
            "    gatk4-4.6.2.0              |  py310hdfd78af_0       380.6 MB  bioconda\n",
            "    giflib-5.2.2               |       hd590300_0          75 KB  conda-forge\n",
            "    graphite2-1.3.14           |       hecca717_2          97 KB  conda-forge\n",
            "    harfbuzz-11.4.5            |       h15599e2_0         2.3 MB  conda-forge\n",
            "    htslib-1.21                |       h5efdd21_0         3.0 MB  bioconda\n",
            "    icu-75.1                   |       he02047a_0        11.6 MB  conda-forge\n",
            "    isa-l-2.31.1               |       hb9d3cd8_1         154 KB  conda-forge\n",
            "    keyutils-1.6.3             |       hb9d3cd8_0         131 KB  conda-forge\n",
            "    krb5-1.21.3                |       h659f571_0         1.3 MB  conda-forge\n",
            "    lcms2-2.17                 |       h717163a_0         242 KB  conda-forge\n",
            "    lerc-4.0.0                 |       h0aef613_1         258 KB  conda-forge\n",
            "    libcups-2.3.3              |       hb8b1518_5         4.3 MB  conda-forge\n",
            "    libcurl-8.17.0             |       h4e3cde8_0         450 KB  conda-forge\n",
            "    libdeflate-1.22            |       hb9d3cd8_0          71 KB  conda-forge\n",
            "    libedit-3.1.20250104       | pl5321h7949ede_0         132 KB  conda-forge\n",
            "    libev-4.33                 |       hd590300_2         110 KB  conda-forge\n",
            "    libexpat-2.7.3             |       hecca717_0          75 KB  conda-forge\n",
            "    libfreetype-2.14.1         |       ha770c72_0           7 KB  conda-forge\n",
            "    libfreetype6-2.14.1        |       h73754d4_0         378 KB  conda-forge\n",
            "    libglib-2.84.4             |       h77a78f3_0         2.8 MB\n",
            "    libiconv-1.18              |       h3b78370_2         772 KB  conda-forge\n",
            "    libjpeg-turbo-3.1.2        |       hb03c661_0         619 KB  conda-forge\n",
            "    libnghttp2-1.67.0          |       had1ee68_0         651 KB  conda-forge\n",
            "    libnsl-2.0.1               |       hb9d3cd8_1          33 KB  conda-forge\n",
            "    libpng-1.6.51              |       h421ea60_0         310 KB  conda-forge\n",
            "    libssh2-1.11.1             |       hcf80075_0         298 KB  conda-forge\n",
            "    libtiff-4.7.0              |       he137b08_1         418 KB  conda-forge\n",
            "    libuuid-2.41.2             |       he9a06e4_0          36 KB  conda-forge\n",
            "    libwebp-base-1.6.0         |       hd42ef1d_0         419 KB  conda-forge\n",
            "    libxcrypt-4.4.36           |       hd590300_1          98 KB  conda-forge\n",
            "    openjdk-17.0.15            |       h5ddf6bc_0       161.8 MB  conda-forge\n",
            "    openssl-3.6.0              |       h26f9b46_0         3.0 MB  conda-forge\n",
            "    pbzip2-1.1.13              |       h1fcc475_2          41 KB  conda-forge\n",
            "    pcre2-10.46                |       h1321c63_0         1.2 MB  conda-forge\n",
            "    perl-5.32.1                | 7_hd590300_perl5        12.7 MB  conda-forge\n",
            "    pigz-2.8                   |       hadc24fc_1          71 KB  conda-forge\n",
            "    pixman-0.46.4              |       h54a6638_1         440 KB  conda-forge\n",
            "    python-3.7.12              |hf930737_100_cpython        57.3 MB  conda-forge\n",
            "    python-isal-1.1.0          |   py37h540881e_0          47 KB  conda-forge\n",
            "    python_abi-3.7             |          4_cp37m           6 KB  conda-forge\n",
            "    samtools-1.21              |       h96c455f_1         476 KB  bioconda\n",
            "    trim-galore-0.6.10         |       hdfd78af_1          45 KB  bioconda\n",
            "    vcftools-0.1.17            | pl5321h077b44d_0         483 KB  bioconda\n",
            "    xopen-1.6.0                |   py37h89c1867_0          27 KB  conda-forge\n",
            "    xorg-libice-1.1.2          |       hb9d3cd8_0          57 KB  conda-forge\n",
            "    xorg-libsm-1.2.6           |       he73a12e_0          27 KB  conda-forge\n",
            "    xorg-libxext-1.3.6         |       hb9d3cd8_0          49 KB  conda-forge\n",
            "    xorg-libxfixes-6.0.2       |       hb03c661_0          20 KB  conda-forge\n",
            "    xorg-libxi-1.8.2           |       hb9d3cd8_0          46 KB  conda-forge\n",
            "    xorg-libxrandr-1.5.4       |       hb9d3cd8_0          29 KB  conda-forge\n",
            "    xorg-libxrender-0.9.12     |       hb9d3cd8_0          32 KB  conda-forge\n",
            "    xorg-libxt-1.3.1           |       hb9d3cd8_0         371 KB  conda-forge\n",
            "    xorg-libxtst-1.2.5         |       hb9d3cd8_3          32 KB  conda-forge\n",
            "    zstd-1.5.7                 |       hb8e6e7a_2         554 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:       667.3 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  alsa-lib           conda-forge/linux-64::alsa-lib-1.2.14-hb9d3cd8_0 \n",
            "  bzip2              conda-forge/linux-64::bzip2-1.0.8-hda65f42_8 \n",
            "  c-ares             conda-forge/linux-64::c-ares-1.34.5-hb9d3cd8_0 \n",
            "  cairo              conda-forge/linux-64::cairo-1.18.4-h3394656_0 \n",
            "  cutadapt           bioconda/linux-64::cutadapt-4.4-py37h8902056_0 \n",
            "  dnaio              bioconda/linux-64::dnaio-0.10.0-py37h8902056_1 \n",
            "  fastqc             bioconda/noarch::fastqc-0.12.1-hdfd78af_0 \n",
            "  font-ttf-dejavu-s~ conda-forge/noarch::font-ttf-dejavu-sans-mono-2.37-hab24e00_0 \n",
            "  font-ttf-inconsol~ conda-forge/noarch::font-ttf-inconsolata-3.000-h77eed37_0 \n",
            "  font-ttf-source-c~ conda-forge/noarch::font-ttf-source-code-pro-2.038-h77eed37_0 \n",
            "  font-ttf-ubuntu    conda-forge/noarch::font-ttf-ubuntu-0.83-h77eed37_3 \n",
            "  fontconfig         conda-forge/linux-64::fontconfig-2.15.0-h7e30c49_1 \n",
            "  fonts-conda-ecosy~ conda-forge/noarch::fonts-conda-ecosystem-1-0 \n",
            "  fonts-conda-forge  conda-forge/noarch::fonts-conda-forge-1-hc364b38_1 \n",
            "  freetype           conda-forge/linux-64::freetype-2.14.1-ha770c72_0 \n",
            "  gatk4              bioconda/noarch::gatk4-4.6.2.0-py310hdfd78af_0 \n",
            "  giflib             conda-forge/linux-64::giflib-5.2.2-hd590300_0 \n",
            "  graphite2          conda-forge/linux-64::graphite2-1.3.14-hecca717_2 \n",
            "  harfbuzz           conda-forge/linux-64::harfbuzz-11.4.5-h15599e2_0 \n",
            "  htslib             bioconda/linux-64::htslib-1.21-h5efdd21_0 \n",
            "  icu                conda-forge/linux-64::icu-75.1-he02047a_0 \n",
            "  isa-l              conda-forge/linux-64::isa-l-2.31.1-hb9d3cd8_1 \n",
            "  keyutils           conda-forge/linux-64::keyutils-1.6.3-hb9d3cd8_0 \n",
            "  krb5               conda-forge/linux-64::krb5-1.21.3-h659f571_0 \n",
            "  lcms2              conda-forge/linux-64::lcms2-2.17-h717163a_0 \n",
            "  lerc               conda-forge/linux-64::lerc-4.0.0-h0aef613_1 \n",
            "  libcups            conda-forge/linux-64::libcups-2.3.3-hb8b1518_5 \n",
            "  libcurl            conda-forge/linux-64::libcurl-8.17.0-h4e3cde8_0 \n",
            "  libdeflate         conda-forge/linux-64::libdeflate-1.22-hb9d3cd8_0 \n",
            "  libedit            conda-forge/linux-64::libedit-3.1.20250104-pl5321h7949ede_0 \n",
            "  libev              conda-forge/linux-64::libev-4.33-hd590300_2 \n",
            "  libexpat           conda-forge/linux-64::libexpat-2.7.3-hecca717_0 \n",
            "  libfreetype        conda-forge/linux-64::libfreetype-2.14.1-ha770c72_0 \n",
            "  libfreetype6       conda-forge/linux-64::libfreetype6-2.14.1-h73754d4_0 \n",
            "  libglib            pkgs/main/linux-64::libglib-2.84.4-h77a78f3_0 \n",
            "  libiconv           conda-forge/linux-64::libiconv-1.18-h3b78370_2 \n",
            "  libjpeg-turbo      conda-forge/linux-64::libjpeg-turbo-3.1.2-hb03c661_0 \n",
            "  libnghttp2         conda-forge/linux-64::libnghttp2-1.67.0-had1ee68_0 \n",
            "  libnsl             conda-forge/linux-64::libnsl-2.0.1-hb9d3cd8_1 \n",
            "  libpng             conda-forge/linux-64::libpng-1.6.51-h421ea60_0 \n",
            "  libssh2            conda-forge/linux-64::libssh2-1.11.1-hcf80075_0 \n",
            "  libtiff            conda-forge/linux-64::libtiff-4.7.0-he137b08_1 \n",
            "  libuuid            conda-forge/linux-64::libuuid-2.41.2-he9a06e4_0 \n",
            "  libwebp-base       conda-forge/linux-64::libwebp-base-1.6.0-hd42ef1d_0 \n",
            "  libxcrypt          conda-forge/linux-64::libxcrypt-4.4.36-hd590300_1 \n",
            "  openjdk            conda-forge/linux-64::openjdk-17.0.15-h5ddf6bc_0 \n",
            "  pbzip2             conda-forge/linux-64::pbzip2-1.1.13-h1fcc475_2 \n",
            "  pcre2              conda-forge/linux-64::pcre2-10.46-h1321c63_0 \n",
            "  perl               conda-forge/linux-64::perl-5.32.1-7_hd590300_perl5 \n",
            "  pigz               conda-forge/linux-64::pigz-2.8-hadc24fc_1 \n",
            "  pixman             conda-forge/linux-64::pixman-0.46.4-h54a6638_1 \n",
            "  python-isal        conda-forge/linux-64::python-isal-1.1.0-py37h540881e_0 \n",
            "  python_abi         conda-forge/linux-64::python_abi-3.7-4_cp37m \n",
            "  samtools           bioconda/linux-64::samtools-1.21-h96c455f_1 \n",
            "  trim-galore        bioconda/noarch::trim-galore-0.6.10-hdfd78af_1 \n",
            "  vcftools           bioconda/linux-64::vcftools-0.1.17-pl5321h077b44d_0 \n",
            "  xopen              conda-forge/linux-64::xopen-1.6.0-py37h89c1867_0 \n",
            "  xorg-libice        conda-forge/linux-64::xorg-libice-1.1.2-hb9d3cd8_0 \n",
            "  xorg-libsm         conda-forge/linux-64::xorg-libsm-1.2.6-he73a12e_0 \n",
            "  xorg-libxext       conda-forge/linux-64::xorg-libxext-1.3.6-hb9d3cd8_0 \n",
            "  xorg-libxfixes     conda-forge/linux-64::xorg-libxfixes-6.0.2-hb03c661_0 \n",
            "  xorg-libxi         conda-forge/linux-64::xorg-libxi-1.8.2-hb9d3cd8_0 \n",
            "  xorg-libxrandr     conda-forge/linux-64::xorg-libxrandr-1.5.4-hb9d3cd8_0 \n",
            "  xorg-libxrender    conda-forge/linux-64::xorg-libxrender-0.9.12-hb9d3cd8_0 \n",
            "  xorg-libxt         conda-forge/linux-64::xorg-libxt-1.3.1-hb9d3cd8_0 \n",
            "  xorg-libxtst       conda-forge/linux-64::xorg-libxtst-1.2.5-hb9d3cd8_3 \n",
            "  zstd               conda-forge/linux-64::zstd-1.5.7-hb8e6e7a_2 \n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates    pkgs/main/linux-64::ca-certificates-2~ --> conda-forge/noarch::ca-certificates-2025.11.12-hbd8a1cb_0 \n",
            "  certifi            pkgs/main/linux-64::certifi-2022.12.7~ --> conda-forge/noarch::certifi-2024.8.30-pyhd8ed1ab_0 \n",
            "  openssl              pkgs/main::openssl-1.1.1w-h7f8727e_0 --> conda-forge::openssl-3.6.0-h26f9b46_0 \n",
            "\n",
            "The following packages will be SUPERSEDED by a higher-priority channel:\n",
            "\n",
            "  python                pkgs/main::python-3.7.16-h7a1cb2a_0 --> conda-forge::python-3.7.12-hf930737_100_cpython \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   0% 0/1 [00:00<?, ?it/s]\n",
            "openjdk-17.0.15      | 161.8 MB  | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 12.7 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "icu-75.1             | 11.6 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "fastqc-0.12.1        | 11.1 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "htslib-1.21          | 3.0 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.6.0        | 3.0 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.4       | 2.8 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.4.5      | 2.3 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-ubuntu-0.83 | 1.5 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "krb5-1.21.3          | 1.3 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "pcre2-10.46          | 1.2 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "cairo-1.18.4         | 955 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libiconv-1.18        | 772 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-source-code | 684 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libnghttp2-1.67.0    | 651 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libjpeg-turbo-3.1.2  | 619 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :   0% 0.00027256019984598284/1 [00:00<13:24, 804.83s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   0% 4.10536134257203e-05/1 [00:00<1:38:49, 5930.19s/it]\n",
            "openjdk-17.0.15      | 161.8 MB  | :   0% 9.655211990863403e-05/1 [00:00<44:26, 2666.39s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "icu-75.1             | 11.6 MB   | :   0% 0.00135078949540213/1 [00:00<03:10, 191.17s/it]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :   7% 0.0746814947577993/1 [00:00<00:03,  3.46s/it]     \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   1% 0.008744419659678425/1 [00:00<00:30, 31.12s/it]    \n",
            "openjdk-17.0.15      | 161.8 MB  | :   2% 0.019793184581269975/1 [00:00<00:14, 14.31s/it]   \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "icu-75.1             | 11.6 MB   | :  30% 0.3039276364654792/1 [00:00<00:00,  1.08it/s]  \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  13% 0.12837585412745792/1 [00:00<00:02,  2.65s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   2% 0.02060891393971159/1 [00:00<00:15, 16.31s/it] \n",
            "openjdk-17.0.15      | 161.8 MB  | :   5% 0.0516553841511192/1 [00:00<00:06,  6.57s/it]  \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "icu-75.1             | 11.6 MB   | :  66% 0.6605360632516415/1 [00:00<00:00,  1.91it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  18% 0.1815250930974246/1 [00:00<00:01,  2.33s/it] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   3% 0.03308921242113056/1 [00:00<00:11, 12.19s/it]\n",
            "openjdk-17.0.15      | 161.8 MB  | :   7% 0.07424858020973957/1 [00:00<00:05,  5.68s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "icu-75.1             | 11.6 MB   | :  96% 0.956358962744708/1 [00:00<00:00,  2.23it/s] \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   4% 0.044994760314589446/1 [00:00<00:10, 10.66s/it]\n",
            "openjdk-17.0.15      | 161.8 MB  | :  10% 0.0979038495873549/1 [00:00<00:04,  5.20s/it] \u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  31% 0.31099118802426645/1 [00:00<00:01,  1.80s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "icu-75.1             | 11.6 MB   | : 100% 1.0/1 [00:00<00:00,  2.23it/s]              \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   6% 0.055545538964999565/1 [00:00<00:10, 11.05s/it]\n",
            "\n",
            "\n",
            "perl-5.32.1          | 12.7 MB   | : 100% 1.0/1 [00:00<00:00,  2.35it/s]               \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "fastqc-0.12.1        | 11.1 MB   | :   0% 0.0014046288797150209/1 [00:00<09:36, 577.44s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | :   0% 0.0036218772527583545/1 [00:00<03:51, 232.39s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  37% 0.3704093115906907/1 [00:00<00:01,  1.98s/it] \u001b[A\u001b[A\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   7% 0.06539840618717244/1 [00:00<00:11, 12.16s/it] \n",
            "\n",
            "\n",
            "\n",
            "\n",
            "fastqc-0.12.1        | 11.1 MB   | :  23% 0.2289545073935484/1 [00:00<00:02,  2.92s/it]    \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | :  57% 0.5650128514303033/1 [00:00<00:00,  1.22s/it]    \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  42% 0.4243762311601953/1 [00:00<00:01,  2.07s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "fastqc-0.12.1        | 11.1 MB   | :  53% 0.5281404587728479/1 [00:01<00:00,  1.24s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   7% 0.07426598668712803/1 [00:01<00:11, 12.87s/it]\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  51% 0.5066894115136822/1 [00:01<00:00,  1.73s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   8% 0.0847346581106867/1 [00:01<00:10, 11.80s/it] \n",
            "openjdk-17.0.15      | 161.8 MB  | :  18% 0.18489730962503415/1 [00:01<00:04,  5.68s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.21it/s]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libcups-2.3.3        | 4.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.21it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  57% 0.57264897987641/1 [00:01<00:00,  1.67s/it]  \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :   9% 0.093643292224068/1 [00:01<00:10, 11.90s/it] \n",
            "openjdk-17.0.15      | 161.8 MB  | :  20% 0.20314566028776598/1 [00:01<00:04,  5.90s/it]\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  63% 0.6347927054412941/1 [00:01<00:00,  1.74s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "htslib-1.21          | 3.0 MB    | :  81% 0.807043646408142/1 [00:01<00:00,  1.17s/it]  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  10% 0.1023466582703207/1 [00:01<00:11, 12.50s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "htslib-1.21          | 3.0 MB    | : 100% 1.0/1 [00:01<00:00,  1.17s/it]              \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  69% 0.6939382688078723/1 [00:01<00:00,  1.85s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  11% 0.11088580986287053/1 [00:01<00:10, 12.29s/it]\n",
            "openjdk-17.0.15      | 161.8 MB  | :  24% 0.2432147900498491/1 [00:01<00:04,  6.13s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "fastqc-0.12.1        | 11.1 MB   | : 100% 1.0/1 [00:01<00:00,  1.20s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "fastqc-0.12.1        | 11.1 MB   | : 100% 1.0/1 [00:01<00:00,  1.20s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  76% 0.761805758569522/1 [00:01<00:00,  1.73s/it] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  12% 0.11921969338829175/1 [00:01<00:11, 13.16s/it]\n",
            "openjdk-17.0.15      | 161.8 MB  | :  26% 0.2603045152736773/1 [00:01<00:04,  6.20s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.6.0        | 3.0 MB    | : 100% 1.0/1 [00:01<00:00,  1.24s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  13% 0.1270198799391786/1 [00:01<00:11, 13.17s/it] \n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.4.5      | 2.3 MB    | :   1% 0.00681973894851813/1 [00:01<04:08, 250.28s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-17.0.15      | 161.8 MB  | :  28% 0.28202874225311997/1 [00:01<00:04,  5.68s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.4       | 2.8 MB    | :   1% 0.00551151197552921/1 [00:01<05:11, 312.81s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  13% 0.13494322733034264/1 [00:01<00:11, 13.02s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.4       | 2.8 MB    | :  85% 0.8487728442314983/1 [00:01<00:00,  1.54s/it]  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-17.0.15      | 161.8 MB  | :  30% 0.3001805407959432/1 [00:01<00:04,  6.21s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.4.5      | 2.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.34s/it]                 \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.4.5      | 2.3 MB    | : 100% 1.0/1 [00:01<00:00,  1.34s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  14% 0.14327711085576386/1 [00:01<00:10, 12.82s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.4       | 2.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.54s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-ubuntu-0.83 | 1.5 MB    | :   1% 0.010110434778315882/1 [00:01<03:06, 188.46s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-17.0.15      | 161.8 MB  | :  32% 0.32103579869620813/1 [00:01<00:03,  5.79s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "krb5-1.21.3          | 1.3 MB    | :   1% 0.011958923317345767/1 [00:01<02:41, 163.16s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | :  99% 0.9885758448413798/1 [00:01<00:00,  1.89s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  15% 0.15115940463350214/1 [00:02<00:10, 12.81s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "krb5-1.21.3          | 1.3 MB    | : 100% 1.0/1 [00:02<00:00, 163.16s/it]                 \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-17.0.15      | 161.8 MB  | :  34% 0.34208416083629034/1 [00:02<00:03,  5.47s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  16% 0.16228493387187234/1 [00:02<00:09, 11.42s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "cairo-1.18.4         | 955 KB    | :   2% 0.016750603712859646/1 [00:02<02:03, 125.57s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "pcre2-10.46          | 1.2 MB    | : 100% 1.0/1 [00:02<00:00, 153.12s/it]                 \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-17.0.15      | 161.8 MB  | :  36% 0.36400149205555027/1 [00:02<00:03,  5.19s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "cairo-1.18.4         | 955 KB    | : 100% 1.0/1 [00:02<00:00, 125.57s/it]                 \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  17% 0.17111146075840222/1 [00:02<00:09, 11.84s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-source-code | 684 KB    | :   2% 0.02337852839697837/1 [00:02<01:32, 95.10s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "openjdk-17.0.15      | 161.8 MB  | :  39% 0.3899740123109728/1 [00:02<00:02,  4.73s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libiconv-1.18        | 772 KB    | : 100% 1.0/1 [00:02<00:00, 105.95s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-source-code | 684 KB    | : 100% 1.0/1 [00:02<00:00, 95.10s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libjpeg-turbo-3.1.2  | 619 KB    | :   3% 0.025854097300026826/1 [00:02<01:27, 89.98s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libnghttp2-1.67.0    | 651 KB    | :   2% 0.02457845784578458/1 [00:02<01:32, 94.81s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  18% 0.17965061235095203/1 [00:02<00:11, 13.97s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "python-3.7.12        | 57.3 MB   | : 100% 1.0/1 [00:02<00:00,  1.89s/it]               \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libjpeg-turbo-3.1.2  | 619 KB    | :  57% 0.5687901406005902/1 [00:02<00:01,  3.10s/it]  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libjpeg-turbo-3.1.2  | 619 KB    | :  88% 0.8790393082009121/1 [00:02<00:00,  1.87s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libjpeg-turbo-3.1.2  | 619 KB    | : 100% 1.0/1 [00:02<00:00,  1.87s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  19% 0.18720447722128458/1 [00:03<00:25, 31.87s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  20% 0.1993974004087235/1 [00:03<00:18, 22.87s/it] \n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  22% 0.22029368964241514/1 [00:03<00:12, 15.91s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  23% 0.23063920022569664/1 [00:03<00:10, 13.99s/it]\n",
            "openjdk-17.0.15      | 161.8 MB  | :  52% 0.5234090420247051/1 [00:03<00:02,  5.84s/it] \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  24% 0.24040996022101807/1 [00:03<00:09, 12.90s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  25% 0.2499343985357852/1 [00:03<00:09, 12.23s/it] \n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  26% 0.25941778323712655/1 [00:03<00:08, 12.05s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  27% 0.27029699079494246/1 [00:03<00:08, 11.13s/it]\n",
            "openjdk-17.0.15      | 161.8 MB  | :  63% 0.6267198103269435/1 [00:03<00:01,  4.44s/it]\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  28% 0.2847068091073703/1 [00:03<00:06,  9.57s/it] \n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  30% 0.29636603532027483/1 [00:04<00:06,  9.27s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  31% 0.31221273010260286/1 [00:04<00:05,  8.17s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  33% 0.32990683748908833/1 [00:04<00:04,  7.24s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  35% 0.34850412437093964/1 [00:04<00:04,  6.57s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  37% 0.3657876956231679/1 [00:04<00:04,  6.32s/it] \n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  38% 0.38278389158141607/1 [00:04<00:03,  6.24s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  40% 0.3989590152711499/1 [00:04<00:03,  6.45s/it] \n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  42% 0.41542151425486373/1 [00:04<00:03,  6.34s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  43% 0.43245876382653764/1 [00:04<00:03,  6.21s/it]\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  46% 0.4647269039791538/1 [00:05<00:03,  6.34s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  86% 0.861427970511889/1 [00:07<00:00,  5.00s/it] \n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  89% 0.8870454252895386/1 [00:07<00:00,  4.64s/it]\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  97% 0.9706716358377308/1 [00:07<00:00,  4.17s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | :  99% 0.9948111605320543/1 [00:07<00:00,  4.30s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "harfbuzz-11.4.5      | 2.3 MB    | : 100% 1.0/1 [00:07<00:00,  1.34s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "fastqc-0.12.1        | 11.1 MB   | : 100% 1.0/1 [00:07<00:00,  1.20s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-ubuntu-0.83 | 1.5 MB    | : 100% 1.0/1 [00:07<00:00,  7.34s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-ubuntu-0.83 | 1.5 MB    | : 100% 1.0/1 [00:07<00:00,  7.34s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libglib-2.84.4       | 2.8 MB    | : 100% 1.0/1 [00:07<00:00,  1.54s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "krb5-1.21.3          | 1.3 MB    | : 100% 1.0/1 [00:08<00:00,  7.53s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "krb5-1.21.3          | 1.3 MB    | : 100% 1.0/1 [00:08<00:00,  7.53s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "cairo-1.18.4         | 955 KB    | : 100% 1.0/1 [00:08<00:00,  7.57s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "cairo-1.18.4         | 955 KB    | : 100% 1.0/1 [00:08<00:00,  7.57s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "pcre2-10.46          | 1.2 MB    | : 100% 1.0/1 [00:08<00:00,  7.63s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "pcre2-10.46          | 1.2 MB    | : 100% 1.0/1 [00:08<00:00,  7.63s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libiconv-1.18        | 772 KB    | : 100% 1.0/1 [00:08<00:00,  7.65s/it] \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libiconv-1.18        | 772 KB    | : 100% 1.0/1 [00:08<00:00,  7.65s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libnghttp2-1.67.0    | 651 KB    | : 100% 1.0/1 [00:08<00:00,  7.67s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libnghttp2-1.67.0    | 651 KB    | : 100% 1.0/1 [00:08<00:00,  7.67s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            " ... (more hidden) ...\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-source-code | 684 KB    | : 100% 1.0/1 [00:08<00:00,  7.82s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "font-ttf-source-code | 684 KB    | : 100% 1.0/1 [00:08<00:00,  7.82s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | : 100% 1.0/1 [00:11<00:00,  4.30s/it]               \n",
            "openjdk-17.0.15      | 161.8 MB  | : 100% 1.0/1 [00:14<00:00,  2.73s/it]\u001b[A\n",
            "\n",
            "gatk4-4.6.2.0        | 380.6 MB  | : 100% 1.0/1 [00:50<00:00,  4.30s/it]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\u001b[A\n",
            "\n",
            "\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\u001b[A\n",
            "\n",
            "\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\u001b[A\n",
            "\n",
            "\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: - \b\b\\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "                        Quality-/Adapter-/RRBS-/Speciality-Trimming\n",
            "                                [powered by Cutadapt]\n",
            "                                  version 0.6.10\n",
            "\n",
            "                               Last update: 02 02 2023\n",
            "\n",
            "samtools: /usr/local/miniconda/envs/workshop_ngs/bin/../lib/libtinfow.so.6: no version information available (required by samtools)\n",
            "samtools: /usr/local/miniconda/envs/workshop_ngs/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
            "samtools: /usr/local/miniconda/envs/workshop_ngs/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
            "samtools: /usr/local/miniconda/envs/workshop_ngs/bin/../lib/libncursesw.so.6: no version information available (required by samtools)\n",
            "samtools 1.21\n",
            "Using htslib 1.21\n",
            "Copyright (C) 2024 Genome Research Ltd.\n",
            "\n",
            "Samtools compilation details:\n",
            "    Features:       build=configure curses=yes \n",
            "    CC:             /opt/conda/conda-bld/samtools_1734249775320/_build_env/bin/x86_64-conda-linux-gnu-cc\n",
            "    CPPFLAGS:       -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /usr/local/miniconda/envs/workshop_ngs/include\n",
            "    CFLAGS:         -Wall -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /usr/local/miniconda/envs/workshop_ngs/include -fdebug-prefix-map=/opt/conda/conda-bld/samtools_1734249775320/work=/usr/local/src/conda/samtools-1.21 -fdebug-prefix-map=/usr/local/miniconda/envs/workshop_ngs=/usr/local/src/conda-prefix\n",
            "    LDFLAGS:        -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/usr/local/miniconda/envs/workshop_ngs/lib -Wl,-rpath-link,/usr/local/miniconda/envs/workshop_ngs/lib -L/usr/local/miniconda/envs/workshop_ngs/lib\n",
            "    HTSDIR:         \n",
            "    LIBS:           \n",
            "    CURSES_LIB:     -ltinfow -lncursesw\n",
            "\n",
            "HTSlib compilation details:\n",
            "    Features:       build=configure libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes plugins=yes plugin-path=/usr/local/miniconda/envs/workshop_ngs/libexec/htslib htscodecs=1.6.1\n",
            "    CC:             /opt/conda/conda-bld/htslib_1726172129301/_build_env/bin/x86_64-conda-linux-gnu-cc\n",
            "    CPPFLAGS:       -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /usr/local/miniconda/envs/workshop_ngs/include\n",
            "    CFLAGS:         -Wall -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /usr/local/miniconda/envs/workshop_ngs/include -fdebug-prefix-map=/opt/conda/conda-bld/htslib_1726172129301/work=/usr/local/src/conda/htslib-1.21 -fdebug-prefix-map=/usr/local/miniconda/envs/workshop_ngs=/usr/local/src/conda-prefix -fvisibility=hidden\n",
            "    LDFLAGS:        -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/usr/local/miniconda/envs/workshop_ngs/lib -Wl,-rpath-link,/usr/local/miniconda/envs/workshop_ngs/lib -L/usr/local/miniconda/envs/workshop_ngs/lib -fvisibility=hidden -rdynamic\n",
            "\n",
            "HTSlib URL scheme handlers present:\n",
            "    built-in:\t file, preload, data\n",
            "    Amazon S3:\t s3+https, s3, s3+http\n",
            "    libcurl:\t gophers, smtp, wss, smb, rtsp, tftp, pop3, smbs, imaps, pop3s, ws, ftps, https, ftp, gopher, sftp, imap, http, smtps, scp, dict, mqtt, telnet\n",
            "    S3 Multipart Upload:\t s3w+https, s3w+http, s3w\n",
            "    Google Cloud Storage:\t gs+http, gs+https, gs\n",
            "    crypt4gh-needed:\t crypt4gh\n",
            "    mem:\t mem\n",
            "FastQC v0.12.1\n",
            "VCFtools (0.1.17)\n",
            "Using GATK jar /usr/local/miniconda/envs/workshop_ngs/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar\n",
            "Running:\n",
            "    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -jar /usr/local/miniconda/envs/workshop_ngs/share/gatk4-4.6.2.0-0/gatk-package-4.6.2.0-local.jar --version\n",
            "The Genome Analysis Toolkit (GATK) v4.6.2.0\n",
            "HTSJDK Version: 4.2.0\n",
            "Picard Version: 3.4.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "70edf711"
      },
      "source": [
        "To activate the `workshop_ngs` environment for specific commands and ensure its tools are used, you can wrap your commands within a `bash -c \"source /usr/local/miniconda/bin/activate workshop_ngs && your_command_here\"` block. This ensures the environment is properly sourced before the command runs. Let's try listing the environment contents and checking `CONDA_PREFIX` within that activated session:"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "e2f52e3b",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "960ae24d-1077-4d52-cc73-d12e76b639a4"
      },
      "source": [
        "!bash -c \"source /usr/local/miniconda/etc/profile.d/conda.sh && conda activate workshop_ngs && \\\n",
        "          echo '--- Environment activated ---' && \\\n",
        "          echo 'CONDA_PREFIX in activated env: '$CONDA_PREFIX && \\\n",
        "          echo '--- Listing contents of activated env ---' && \\\n",
        "          conda list\""
      ],
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--- Environment activated ---\n",
            "CONDA_PREFIX in activated env: \n",
            "--- Listing contents of activated env ---\n",
            "# packages in environment at /usr/local/miniconda/envs/workshop_ngs:\n",
            "#\n",
            "# Name                       Version          Build                 Channel\n",
            "_libgcc_mutex                0.1              main\n",
            "_openmp_mutex                5.1              1_gnu\n",
            "alsa-lib                     1.2.14           hb9d3cd8_0            conda-forge\n",
            "bzip2                        1.0.8            hda65f42_8            conda-forge\n",
            "c-ares                       1.34.5           hb9d3cd8_0            conda-forge\n",
            "ca-certificates              2025.11.12       hbd8a1cb_0            conda-forge\n",
            "cairo                        1.18.4           h3394656_0            conda-forge\n",
            "certifi                      2024.8.30        pyhd8ed1ab_0          conda-forge\n",
            "cutadapt                     4.4              py37h8902056_0        bioconda\n",
            "dnaio                        0.10.0           py37h8902056_1        bioconda\n",
            "fastqc                       0.12.1           hdfd78af_0            bioconda\n",
            "font-ttf-dejavu-sans-mono    2.37             hab24e00_0            conda-forge\n",
            "font-ttf-inconsolata         3.000            h77eed37_0            conda-forge\n",
            "font-ttf-source-code-pro     2.038            h77eed37_0            conda-forge\n",
            "font-ttf-ubuntu              0.83             h77eed37_3            conda-forge\n",
            "fontconfig                   2.15.0           h7e30c49_1            conda-forge\n",
            "fonts-conda-ecosystem        1                0                     conda-forge\n",
            "fonts-conda-forge            1                hc364b38_1            conda-forge\n",
            "freetype                     2.14.1           ha770c72_0            conda-forge\n",
            "gatk4                        4.6.2.0          py310hdfd78af_0       bioconda\n",
            "giflib                       5.2.2            hd590300_0            conda-forge\n",
            "graphite2                    1.3.14           hecca717_2            conda-forge\n",
            "harfbuzz                     11.4.5           h15599e2_0            conda-forge\n",
            "htslib                       1.21             h5efdd21_0            bioconda\n",
            "icu                          75.1             he02047a_0            conda-forge\n",
            "isa-l                        2.31.1           hb9d3cd8_1            conda-forge\n",
            "keyutils                     1.6.3            hb9d3cd8_0            conda-forge\n",
            "krb5                         1.21.3           h659f571_0            conda-forge\n",
            "lcms2                        2.17             h717163a_0            conda-forge\n",
            "ld_impl_linux-64             2.44             h153f514_2\n",
            "lerc                         4.0.0            h0aef613_1            conda-forge\n",
            "libcups                      2.3.3            hb8b1518_5            conda-forge\n",
            "libcurl                      8.17.0           h4e3cde8_0            conda-forge\n",
            "libdeflate                   1.22             hb9d3cd8_0            conda-forge\n",
            "libedit                      3.1.20250104     pl5321h7949ede_0      conda-forge\n",
            "libev                        4.33             hd590300_2            conda-forge\n",
            "libexpat                     2.7.3            hecca717_0            conda-forge\n",
            "libffi                       3.4.4            h6a678d5_1\n",
            "libfreetype                  2.14.1           ha770c72_0            conda-forge\n",
            "libfreetype6                 2.14.1           h73754d4_0            conda-forge\n",
            "libgcc                       15.2.0           h69a1729_7\n",
            "libgcc-ng                    15.2.0           h166f726_7\n",
            "libglib                      2.84.4           h77a78f3_0\n",
            "libgomp                      15.2.0           h4751f2c_7\n",
            "libiconv                     1.18             h3b78370_2            conda-forge\n",
            "libjpeg-turbo                3.1.2            hb03c661_0            conda-forge\n",
            "libnghttp2                   1.67.0           had1ee68_0            conda-forge\n",
            "libnsl                       2.0.1            hb9d3cd8_1            conda-forge\n",
            "libpng                       1.6.51           h421ea60_0            conda-forge\n",
            "libssh2                      1.11.1           hcf80075_0            conda-forge\n",
            "libstdcxx                    15.2.0           h39759b7_7\n",
            "libstdcxx-ng                 15.2.0           hc03a8fd_7\n",
            "libtiff                      4.7.0            he137b08_1            conda-forge\n",
            "libuuid                      2.41.2           he9a06e4_0            conda-forge\n",
            "libwebp-base                 1.6.0            hd42ef1d_0            conda-forge\n",
            "libxcb                       1.17.0           h9b100fa_0\n",
            "libxcrypt                    4.4.36           hd590300_1            conda-forge\n",
            "libzlib                      1.3.1            hb25bd0a_0\n",
            "ncurses                      6.5              h7934f7d_0\n",
            "openjdk                      17.0.15          h5ddf6bc_0            conda-forge\n",
            "openssl                      3.6.0            h26f9b46_0            conda-forge\n",
            "pbzip2                       1.1.13           h1fcc475_2            conda-forge\n",
            "pcre2                        10.46            h1321c63_0            conda-forge\n",
            "perl                         5.32.1           7_hd590300_perl5      conda-forge\n",
            "pigz                         2.8              hadc24fc_1            conda-forge\n",
            "pip                          22.3.1           py37h06a4308_0\n",
            "pixman                       0.46.4           h54a6638_1            conda-forge\n",
            "pthread-stubs                0.3              h0ce48e5_1\n",
            "python                       3.7.12           hf930737_100_cpython  conda-forge\n",
            "python-isal                  1.1.0            py37h540881e_0        conda-forge\n",
            "python_abi                   3.7              4_cp37m               conda-forge\n",
            "readline                     8.3              hc2a1206_0\n",
            "samtools                     1.21             h96c455f_1            bioconda\n",
            "setuptools                   65.6.3           py37h06a4308_0\n",
            "sqlite                       3.51.0           h2a70700_0\n",
            "tk                           8.6.15           h54e0aa7_0\n",
            "trim-galore                  0.6.10           hdfd78af_1            bioconda\n",
            "vcftools                     0.1.17           pl5321h077b44d_0      bioconda\n",
            "wheel                        0.38.4           py37h06a4308_0\n",
            "xopen                        1.6.0            py37h89c1867_0        conda-forge\n",
            "xorg-libice                  1.1.2            hb9d3cd8_0            conda-forge\n",
            "xorg-libsm                   1.2.6            he73a12e_0            conda-forge\n",
            "xorg-libx11                  1.8.12           h9b100fa_1\n",
            "xorg-libxau                  1.0.12           h9b100fa_0\n",
            "xorg-libxdmcp                1.1.5            h9b100fa_0\n",
            "xorg-libxext                 1.3.6            hb9d3cd8_0            conda-forge\n",
            "xorg-libxfixes               6.0.2            hb03c661_0            conda-forge\n",
            "xorg-libxi                   1.8.2            hb9d3cd8_0            conda-forge\n",
            "xorg-libxrandr               1.5.4            hb9d3cd8_0            conda-forge\n",
            "xorg-libxrender              0.9.12           hb9d3cd8_0            conda-forge\n",
            "xorg-libxt                   1.3.1            hb9d3cd8_0            conda-forge\n",
            "xorg-libxtst                 1.2.5            hb9d3cd8_3            conda-forge\n",
            "xorg-xorgproto               2024.1           h5eee18b_1\n",
            "xz                           5.6.4            h5eee18b_1\n",
            "zlib                         1.3.1            hb25bd0a_0\n",
            "zstd                         1.5.7            hb8e6e7a_2            conda-forge\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5c5c06e3"
      },
      "source": [
        "We’ll be using real WGS resequencing data for this workshop. The dataset is a small subset for demonstration purposes. We will download the data from (https://zenodo.org/records/14258052) . Step 1 would be to make a parent directory to sort all the data in one place.\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f0a0fc21",
        "outputId": "de8da757-abd1-4460-90ba-52ecd34dfabd"
      },
      "source": [
        "!mkdir -p fastq_files\n",
        "!ls -F"
      ],
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "fastq_files/  miniconda.sh  sample_data/\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "58d954b7"
      },
      "source": [
        "Now, let's navigate into the `fastq_files` directory."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fd7e272d",
        "outputId": "238089f2-fe4d-4a38-a6d1-023d7c098321"
      },
      "source": [
        "!cd fastq_files/ && pwd"
      ],
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/fastq_files\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "8d6ca911"
      },
      "source": [
        "Let's organize the download process by first ensuring the `fastq_files` directory exists, then navigating into it to download all the FASTQ files directly to their intended location."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bb6bbeb7",
        "outputId": "cc419ad5-98b8-4517-e79e-e3cf773a72fa"
      },
      "source": [
        "# Define the list of FASTQ file links\n",
        "fastq_links = [\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_CI16_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_CI16_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_CI18_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_CI18_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_NW10_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_NW10_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_NW12_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_NW12_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_NW13_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_NW13_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_SI18_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_SI18_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_SI19_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_SI19_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_SI9_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/BEN_SI9_sub_2.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/LGS1_sub_1.fq.gz\",\n",
        "    \"https://zenodo.org/records/14258052/files/LGS1_sub_2.fq.gz\"\n",
        "]\n",
        "\n",
        "# Create the directory if it doesn't exist, and then download files into it\n",
        "!mkdir -p fastq_files\n",
        "\n",
        "# Change to the directory and download files\n",
        "# Using a subshell (bash -c) ensures the 'cd' command applies to the subsequent 'wget' commands\n",
        "for link in fastq_links:\n",
        "    !bash -c \"cd fastq_files && wget -P . -q {link}\"\n",
        "\n",
        "# List the contents of the fastq_files directory to confirm all downloads\n",
        "!ls -F fastq_files/"
      ],
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "BEN_CI16_sub_1.fq.gz\tBEN_NW12_sub_1.fq.gz\tBEN_SI19_sub_1.fq.gz\n",
            "BEN_CI16_sub_1.fq.gz.1\tBEN_NW12_sub_1.fq.gz.1\tBEN_SI19_sub_1.fq.gz.1\n",
            "BEN_CI16_sub_2.fq.gz\tBEN_NW12_sub_2.fq.gz\tBEN_SI19_sub_2.fq.gz\n",
            "BEN_CI16_sub_2.fq.gz.1\tBEN_NW12_sub_2.fq.gz.1\tBEN_SI19_sub_2.fq.gz.1\n",
            "BEN_CI18_sub_1.fq.gz\tBEN_NW13_sub_1.fq.gz\tBEN_SI9_sub_1.fq.gz\n",
            "BEN_CI18_sub_1.fq.gz.1\tBEN_NW13_sub_1.fq.gz.1\tBEN_SI9_sub_1.fq.gz.1\n",
            "BEN_CI18_sub_2.fq.gz\tBEN_NW13_sub_2.fq.gz\tBEN_SI9_sub_2.fq.gz\n",
            "BEN_CI18_sub_2.fq.gz.1\tBEN_NW13_sub_2.fq.gz.1\tBEN_SI9_sub_2.fq.gz.1\n",
            "BEN_NW10_sub_1.fq.gz\tBEN_SI18_sub_1.fq.gz\tLGS1_sub_1.fq.gz\n",
            "BEN_NW10_sub_1.fq.gz.1\tBEN_SI18_sub_1.fq.gz.1\tLGS1_sub_1.fq.gz.1\n",
            "BEN_NW10_sub_2.fq.gz\tBEN_SI18_sub_2.fq.gz\tLGS1_sub_2.fq.gz\n",
            "BEN_NW10_sub_2.fq.gz.1\tBEN_SI18_sub_2.fq.gz.1\tLGS1_sub_2.fq.gz.1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "To visualise the content of the directory 'fastq_files' , to confirm that all the files have been successfully downloaded."
      ],
      "metadata": {
        "id": "NIKJbD1kB49q"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5e74bb29"
      },
      "source": [
        "Once all the fastq files are downloaded, the next would be to run \"FastQC\" tool on the raw reads for the quality assessment step before the downstream analysis part.\n",
        "\n",
        "***Why it's important:*** Poor-quality reads can lead to false variant calls or poor mapping, so quality control is crucial.\n",
        "\n",
        "To do so, first, let's create a directory to store the FastQC reports."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "93be3c96",
        "outputId": "93e5ab2b-3d29-42b2-cbcc-34601993d520"
      },
      "source": [
        "!mkdir -p fastqc_results\n",
        "!ls -F"
      ],
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "fastqc_results/  fastq_files/  miniconda.sh  sample_data/\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1cf4153e"
      },
      "source": [
        "Now, we will run FastQC on all the FASTQ files within the `workshop_ngs` environment and save the reports to the `fastqc_results` directory."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ddc193b1",
        "outputId": "2a4d8bb4-df55-4da8-a5ee-b1ea56f4b4ec"
      },
      "source": [
        "%%time\n",
        "!bash -c \"source /usr/local/miniconda/etc/profile.d/conda.sh && conda activate workshop_ngs && \\\n",
        "          fastqc fastq_files/*.fq.gz -o fastqc_results\"\n",
        "\n",
        "# List the generated FastQC reports\n",
        "!ls -F fastqc_results/"
      ],
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "Started analysis of BEN_CI16_sub_1.fq.gz\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "application/gzip\n",
            "Approx 5% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_CI16_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_CI16_sub_1.fq.gz\n",
            "Analysis complete for BEN_CI16_sub_1.fq.gz\n",
            "Started analysis of BEN_CI16_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_CI16_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_CI16_sub_2.fq.gz\n",
            "Analysis complete for BEN_CI16_sub_2.fq.gz\n",
            "Started analysis of BEN_CI18_sub_1.fq.gz\n",
            "Approx 5% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_CI18_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_CI18_sub_1.fq.gz\n",
            "Analysis complete for BEN_CI18_sub_1.fq.gz\n",
            "Started analysis of BEN_CI18_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_CI18_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_CI18_sub_2.fq.gz\n",
            "Analysis complete for BEN_CI18_sub_2.fq.gz\n",
            "Started analysis of BEN_NW10_sub_1.fq.gz\n",
            "Approx 5% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_NW10_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_NW10_sub_1.fq.gz\n",
            "Analysis complete for BEN_NW10_sub_1.fq.gz\n",
            "Started analysis of BEN_NW10_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_NW10_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_NW10_sub_2.fq.gz\n",
            "Analysis complete for BEN_NW10_sub_2.fq.gz\n",
            "Started analysis of BEN_NW12_sub_1.fq.gz\n",
            "Approx 5% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_NW12_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_NW12_sub_1.fq.gz\n",
            "Analysis complete for BEN_NW12_sub_1.fq.gz\n",
            "Started analysis of BEN_NW12_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_NW12_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_NW12_sub_2.fq.gz\n",
            "Analysis complete for BEN_NW12_sub_2.fq.gz\n",
            "Started analysis of BEN_NW13_sub_1.fq.gz\n",
            "Approx 5% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_NW13_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_NW13_sub_1.fq.gz\n",
            "Analysis complete for BEN_NW13_sub_1.fq.gz\n",
            "Started analysis of BEN_NW13_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_NW13_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_NW13_sub_2.fq.gz\n",
            "Analysis complete for BEN_NW13_sub_2.fq.gz\n",
            "Started analysis of BEN_SI18_sub_1.fq.gz\n",
            "Approx 5% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_SI18_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_SI18_sub_1.fq.gz\n",
            "Analysis complete for BEN_SI18_sub_1.fq.gz\n",
            "Started analysis of BEN_SI18_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_SI18_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_SI18_sub_2.fq.gz\n",
            "Analysis complete for BEN_SI18_sub_2.fq.gz\n",
            "Started analysis of BEN_SI19_sub_1.fq.gz\n",
            "Approx 5% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_SI19_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_SI19_sub_1.fq.gz\n",
            "Analysis complete for BEN_SI19_sub_1.fq.gz\n",
            "Started analysis of BEN_SI19_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_SI19_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_SI19_sub_2.fq.gz\n",
            "Analysis complete for BEN_SI19_sub_2.fq.gz\n",
            "Started analysis of BEN_SI9_sub_1.fq.gz\n",
            "Approx 5% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 10% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 15% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 20% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 25% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 30% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 35% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 40% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 45% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 50% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 55% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 60% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 65% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 70% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 75% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 80% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 85% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 90% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 95% complete for BEN_SI9_sub_1.fq.gz\n",
            "Approx 100% complete for BEN_SI9_sub_1.fq.gz\n",
            "Analysis complete for BEN_SI9_sub_1.fq.gz\n",
            "Started analysis of BEN_SI9_sub_2.fq.gz\n",
            "Approx 5% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 10% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 15% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 20% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 25% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 30% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 35% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 40% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 45% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 50% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 55% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 60% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 65% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 70% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 75% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 80% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 85% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 90% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 95% complete for BEN_SI9_sub_2.fq.gz\n",
            "Approx 100% complete for BEN_SI9_sub_2.fq.gz\n",
            "Analysis complete for BEN_SI9_sub_2.fq.gz\n",
            "Started analysis of LGS1_sub_1.fq.gz\n",
            "Approx 5% complete for LGS1_sub_1.fq.gz\n",
            "Approx 10% complete for LGS1_sub_1.fq.gz\n",
            "Approx 15% complete for LGS1_sub_1.fq.gz\n",
            "Approx 20% complete for LGS1_sub_1.fq.gz\n",
            "Approx 25% complete for LGS1_sub_1.fq.gz\n",
            "Approx 30% complete for LGS1_sub_1.fq.gz\n",
            "Approx 35% complete for LGS1_sub_1.fq.gz\n",
            "Approx 40% complete for LGS1_sub_1.fq.gz\n",
            "Approx 45% complete for LGS1_sub_1.fq.gz\n",
            "Approx 50% complete for LGS1_sub_1.fq.gz\n",
            "Approx 55% complete for LGS1_sub_1.fq.gz\n",
            "Approx 60% complete for LGS1_sub_1.fq.gz\n",
            "Approx 65% complete for LGS1_sub_1.fq.gz\n",
            "Approx 70% complete for LGS1_sub_1.fq.gz\n",
            "Approx 75% complete for LGS1_sub_1.fq.gz\n",
            "Approx 80% complete for LGS1_sub_1.fq.gz\n",
            "Approx 85% complete for LGS1_sub_1.fq.gz\n",
            "Approx 90% complete for LGS1_sub_1.fq.gz\n",
            "Approx 95% complete for LGS1_sub_1.fq.gz\n",
            "Approx 100% complete for LGS1_sub_1.fq.gz\n",
            "Analysis complete for LGS1_sub_1.fq.gz\n",
            "Started analysis of LGS1_sub_2.fq.gz\n",
            "Approx 5% complete for LGS1_sub_2.fq.gz\n",
            "Approx 10% complete for LGS1_sub_2.fq.gz\n",
            "Approx 15% complete for LGS1_sub_2.fq.gz\n",
            "Approx 20% complete for LGS1_sub_2.fq.gz\n",
            "Approx 25% complete for LGS1_sub_2.fq.gz\n",
            "Approx 30% complete for LGS1_sub_2.fq.gz\n",
            "Approx 35% complete for LGS1_sub_2.fq.gz\n",
            "Approx 40% complete for LGS1_sub_2.fq.gz\n",
            "Approx 45% complete for LGS1_sub_2.fq.gz\n",
            "Approx 50% complete for LGS1_sub_2.fq.gz\n",
            "Approx 55% complete for LGS1_sub_2.fq.gz\n",
            "Approx 60% complete for LGS1_sub_2.fq.gz\n",
            "Approx 65% complete for LGS1_sub_2.fq.gz\n",
            "Approx 70% complete for LGS1_sub_2.fq.gz\n",
            "Approx 75% complete for LGS1_sub_2.fq.gz\n",
            "Approx 80% complete for LGS1_sub_2.fq.gz\n",
            "Approx 85% complete for LGS1_sub_2.fq.gz\n",
            "Approx 90% complete for LGS1_sub_2.fq.gz\n",
            "Approx 95% complete for LGS1_sub_2.fq.gz\n",
            "Approx 100% complete for LGS1_sub_2.fq.gz\n",
            "Analysis complete for LGS1_sub_2.fq.gz\n",
            "BEN_CI16_sub_1_fastqc.html  BEN_NW13_sub_2_fastqc.html\n",
            "BEN_CI16_sub_1_fastqc.zip   BEN_NW13_sub_2_fastqc.zip\n",
            "BEN_CI16_sub_2_fastqc.html  BEN_SI18_sub_1_fastqc.html\n",
            "BEN_CI16_sub_2_fastqc.zip   BEN_SI18_sub_1_fastqc.zip\n",
            "BEN_CI18_sub_1_fastqc.html  BEN_SI18_sub_2_fastqc.html\n",
            "BEN_CI18_sub_1_fastqc.zip   BEN_SI18_sub_2_fastqc.zip\n",
            "BEN_CI18_sub_2_fastqc.html  BEN_SI19_sub_1_fastqc.html\n",
            "BEN_CI18_sub_2_fastqc.zip   BEN_SI19_sub_1_fastqc.zip\n",
            "BEN_NW10_sub_1_fastqc.html  BEN_SI19_sub_2_fastqc.html\n",
            "BEN_NW10_sub_1_fastqc.zip   BEN_SI19_sub_2_fastqc.zip\n",
            "BEN_NW10_sub_2_fastqc.html  BEN_SI9_sub_1_fastqc.html\n",
            "BEN_NW10_sub_2_fastqc.zip   BEN_SI9_sub_1_fastqc.zip\n",
            "BEN_NW12_sub_1_fastqc.html  BEN_SI9_sub_2_fastqc.html\n",
            "BEN_NW12_sub_1_fastqc.zip   BEN_SI9_sub_2_fastqc.zip\n",
            "BEN_NW12_sub_2_fastqc.html  LGS1_sub_1_fastqc.html\n",
            "BEN_NW12_sub_2_fastqc.zip   LGS1_sub_1_fastqc.zip\n",
            "BEN_NW13_sub_1_fastqc.html  LGS1_sub_2_fastqc.html\n",
            "BEN_NW13_sub_1_fastqc.zip   LGS1_sub_2_fastqc.zip\n",
            "CPU times: user 575 ms, sys: 80.1 ms, total: 655 ms\n",
            "Wall time: 3min 2s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The .html files generated by FastQC are standard web pages, which means you can visualize them by opening them in any web browser!\n",
        "\n",
        "Here's how you can access them from Colab:\n",
        " On the left-hand side of your Colab interface, there's usually a file explorer icon (a folder). Click on it to open the file browser. Navigate to the fastqc_results directory. You'll see all the .html files listed there. You can right-click on any .html file and select 'Download' to save it to your local machine, then open it with your preferred web browser.\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "TLmb_MoTEJ-p"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "26726bab"
      },
      "source": [
        "Post qualitty assessment, the next step will be the trimming of the adapters or low-quality bases using tools such as Trim-galore. This step will ensure producing clean FASTQ files, ready for mapping.\n",
        "\n",
        "Step 1: We'll create a directory to store the trimmed FASTQ files."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "06e02af7",
        "outputId": "3b5db0e2-4c50-4533-aa25-901bd6eb83c6"
      },
      "source": [
        "!mkdir -p trimmed_fastq_files\n",
        "!ls -F"
      ],
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "fastqc_results/  fastq_files/  miniconda.sh  sample_data/  trimmed_fastq_files/\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9bbe2bd9"
      },
      "source": [
        "Now, we will activate the `workshop_ngs` conda environment and run `Trim Galore!` on all the FASTQ files. We'll specify that they are paired-end reads and direct the output to the `trimmed_fastq_files` directory.\n",
        "\n",
        "Trim Galore automatically detects and removes common adapter sequences and performs quality trimming. The `--paired` option is essential for correctly processing paired-end reads."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f3bd792c",
        "outputId": "36e15009-08d9-4e06-9b73-0c17d416ad3f"
      },
      "source": [
        "%%time\n",
        "# Run Trim Galore! within the activated conda environment\n",
        "# We iterate through the samples to process paired-end reads correctly\n",
        "\n",
        "# Get a list of fastq files\n",
        "fastq_files_list = !ls fastq_files/*.fq.gz\n",
        "\n",
        "# Extract unique sample prefixes (e.g., BEN_CI16, BEN_CI18) from the filenames\n",
        "# We split by '/' to get just the filename, then split by '_sub_' to get the sample prefix\n",
        "sample_prefixes = sorted(list(set([f.split('/')[-1].split('_sub_')[0] for f in fastq_files_list])))\n",
        "\n",
        "for prefix in sample_prefixes:\n",
        "    read1 = f\"fastq_files/{prefix}_sub_1.fq.gz\"\n",
        "    read2 = f\"fastq_files/{prefix}_sub_2.fq.gz\"\n",
        "    output_dir = \"trimmed_fastq_files\"\n",
        "\n",
        "    # Use bash -c to activate conda env and run trim_galore\n",
        "    # --paired: for paired-end reads\n",
        "    # -o: specify output directory\n",
        "    # --fastqc: run FastQC on the trimmed output (optional but good for QC after trimming)\n",
        "    !bash -c \"source /usr/local/miniconda/etc/profile.d/conda.sh && conda activate workshop_ngs && \\\n",
        "              trim_galore --paired {read1} {read2} -o {output_dir} --fastqc\"\n",
        "\n",
        "# List the contents of the trimmed_fastq_files directory to confirm\n",
        "!ls -F trimmed_fastq_files/"
      ],
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Multicore support not enabled. Proceeding with single-core trimming.\n",
            "Path to Cutadapt set as: 'cutadapt' (default)\n",
            "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
            "Cutadapt version: 4.4\n",
            "single-core operation.\n",
            "igzip command line interface 2.31.1\n",
            "igzip detected. Using igzip for decompressing\n",
            "\n",
            "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
            "\n",
            "Output will be written into the directory: /content/trimmed_fastq_files/\n",
            "\n",
            "\n",
            "AUTO-DETECTING ADAPTER TYPE\n",
            "===========================\n",
            "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastq_files/BEN_NW13_sub_1.fq.gz <<)\n",
            "\n",
            "Found perfect matches for the following adapter sequences:\n",
            "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
            "Illumina\t43421\tAGATCGGAAGAGC\t1000000\t4.34\n",
            "Nextera\t14\tCTGTCTCTTATA\t1000000\t0.00\n",
            "smallRNA\t6\tTGGAATTCTCGG\t1000000\t0.00\n",
            "Using Illumina adapter for trimming (count: 43421). Second best hit was Nextera (count: 14)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_NW13_sub_1.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_NW13_sub_1.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_NW13_sub_1_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_NW13_sub_1.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_NW13_sub_1.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 55.939 s (55.939 µs/read; 1.07 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   394,500 (39.5%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   150,000,000 bp\n",
            "Quality-trimmed:                 605,803 bp (0.4%)\n",
            "Total written (filtered):    147,193,131 bp (98.1%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 394500 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 30.2%\n",
            "  C: 27.9%\n",
            "  G: 19.5%\n",
            "  T: 22.3%\n",
            "  none/other: 0.2%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t230852\t250000.0\t0\t230852\n",
            "2\t70334\t62500.0\t0\t70334\n",
            "3\t23260\t15625.0\t0\t23260\n",
            "4\t6433\t3906.2\t0\t6433\n",
            "5\t3176\t976.6\t0\t3176\n",
            "6\t2305\t244.1\t0\t2305\n",
            "7\t2189\t61.0\t0\t2189\n",
            "8\t2159\t15.3\t0\t2159\n",
            "9\t2034\t3.8\t0\t2007 27\n",
            "10\t2020\t1.0\t1\t1924 96\n",
            "11\t1909\t0.2\t1\t1806 103\n",
            "12\t1877\t0.1\t1\t1789 88\n",
            "13\t1790\t0.0\t1\t1701 89\n",
            "14\t1843\t0.0\t1\t1751 92\n",
            "15\t1648\t0.0\t1\t1557 91\n",
            "16\t1652\t0.0\t1\t1563 89\n",
            "17\t1618\t0.0\t1\t1518 100\n",
            "18\t1530\t0.0\t1\t1434 96\n",
            "19\t1354\t0.0\t1\t1280 74\n",
            "20\t1408\t0.0\t1\t1311 97\n",
            "21\t1366\t0.0\t1\t1280 86\n",
            "22\t1361\t0.0\t1\t1277 84\n",
            "23\t1219\t0.0\t1\t1147 72\n",
            "24\t1204\t0.0\t1\t1138 66\n",
            "25\t1158\t0.0\t1\t1104 54\n",
            "26\t1097\t0.0\t1\t1047 50\n",
            "27\t1030\t0.0\t1\t989 41\n",
            "28\t1049\t0.0\t1\t994 55\n",
            "29\t1011\t0.0\t1\t948 63\n",
            "30\t944\t0.0\t1\t908 36\n",
            "31\t897\t0.0\t1\t862 35\n",
            "32\t861\t0.0\t1\t832 29\n",
            "33\t834\t0.0\t1\t814 20\n",
            "34\t839\t0.0\t1\t808 31\n",
            "35\t782\t0.0\t1\t757 25\n",
            "36\t785\t0.0\t1\t749 36\n",
            "37\t736\t0.0\t1\t715 21\n",
            "38\t701\t0.0\t1\t676 25\n",
            "39\t646\t0.0\t1\t622 24\n",
            "40\t633\t0.0\t1\t613 20\n",
            "41\t609\t0.0\t1\t582 27\n",
            "42\t625\t0.0\t1\t607 18\n",
            "43\t633\t0.0\t1\t613 20\n",
            "44\t511\t0.0\t1\t495 16\n",
            "45\t668\t0.0\t1\t647 21\n",
            "46\t282\t0.0\t1\t275 7\n",
            "47\t437\t0.0\t1\t428 9\n",
            "48\t431\t0.0\t1\t425 6\n",
            "49\t417\t0.0\t1\t405 12\n",
            "50\t399\t0.0\t1\t388 11\n",
            "51\t409\t0.0\t1\t397 12\n",
            "52\t365\t0.0\t1\t354 11\n",
            "53\t373\t0.0\t1\t362 11\n",
            "54\t376\t0.0\t1\t363 13\n",
            "55\t347\t0.0\t1\t335 12\n",
            "56\t287\t0.0\t1\t274 13\n",
            "57\t291\t0.0\t1\t283 8\n",
            "58\t315\t0.0\t1\t304 11\n",
            "59\t284\t0.0\t1\t276 8\n",
            "60\t295\t0.0\t1\t288 7\n",
            "61\t250\t0.0\t1\t246 4\n",
            "62\t235\t0.0\t1\t228 7\n",
            "63\t264\t0.0\t1\t254 10\n",
            "64\t230\t0.0\t1\t224 6\n",
            "65\t209\t0.0\t1\t202 7\n",
            "66\t234\t0.0\t1\t219 15\n",
            "67\t207\t0.0\t1\t198 9\n",
            "68\t178\t0.0\t1\t176 2\n",
            "69\t217\t0.0\t1\t211 6\n",
            "70\t216\t0.0\t1\t195 21\n",
            "71\t568\t0.0\t1\t229 339\n",
            "72\t1547\t0.0\t1\t1446 101\n",
            "73\t331\t0.0\t1\t301 30\n",
            "74\t119\t0.0\t1\t95 24\n",
            "75\t49\t0.0\t1\t40 9\n",
            "76\t48\t0.0\t1\t40 8\n",
            "77\t35\t0.0\t1\t29 6\n",
            "78\t29\t0.0\t1\t28 1\n",
            "79\t24\t0.0\t1\t22 2\n",
            "80\t28\t0.0\t1\t23 5\n",
            "81\t20\t0.0\t1\t20\n",
            "82\t22\t0.0\t1\t20 2\n",
            "83\t20\t0.0\t1\t16 4\n",
            "84\t21\t0.0\t1\t18 3\n",
            "85\t19\t0.0\t1\t18 1\n",
            "86\t26\t0.0\t1\t25 1\n",
            "87\t21\t0.0\t1\t19 2\n",
            "88\t16\t0.0\t1\t13 3\n",
            "89\t17\t0.0\t1\t16 1\n",
            "90\t12\t0.0\t1\t11 1\n",
            "91\t13\t0.0\t1\t12 1\n",
            "92\t13\t0.0\t1\t13\n",
            "93\t14\t0.0\t1\t14\n",
            "94\t8\t0.0\t1\t8\n",
            "95\t8\t0.0\t1\t5 3\n",
            "96\t7\t0.0\t1\t5 2\n",
            "97\t9\t0.0\t1\t3 6\n",
            "98\t8\t0.0\t1\t6 2\n",
            "99\t8\t0.0\t1\t4 4\n",
            "100\t7\t0.0\t1\t5 2\n",
            "101\t9\t0.0\t1\t7 2\n",
            "102\t11\t0.0\t1\t10 1\n",
            "103\t8\t0.0\t1\t6 2\n",
            "104\t9\t0.0\t1\t6 3\n",
            "105\t3\t0.0\t1\t3\n",
            "106\t14\t0.0\t1\t10 4\n",
            "107\t8\t0.0\t1\t8\n",
            "108\t5\t0.0\t1\t4 1\n",
            "109\t7\t0.0\t1\t6 1\n",
            "110\t9\t0.0\t1\t6 3\n",
            "111\t5\t0.0\t1\t4 1\n",
            "112\t8\t0.0\t1\t7 1\n",
            "113\t7\t0.0\t1\t7\n",
            "114\t2\t0.0\t1\t2\n",
            "115\t3\t0.0\t1\t3\n",
            "116\t2\t0.0\t1\t1 1\n",
            "117\t9\t0.0\t1\t6 3\n",
            "118\t2\t0.0\t1\t2\n",
            "119\t2\t0.0\t1\t2\n",
            "120\t1\t0.0\t1\t0 1\n",
            "121\t3\t0.0\t1\t2 1\n",
            "123\t1\t0.0\t1\t1\n",
            "125\t4\t0.0\t1\t1 3\n",
            "126\t3\t0.0\t1\t1 2\n",
            "127\t1\t0.0\t1\t0 1\n",
            "128\t2\t0.0\t1\t0 2\n",
            "130\t3\t0.0\t1\t0 3\n",
            "131\t1\t0.0\t1\t0 1\n",
            "132\t2\t0.0\t1\t0 2\n",
            "135\t1\t0.0\t1\t0 1\n",
            "136\t2\t0.0\t1\t0 2\n",
            "137\t2\t0.0\t1\t0 2\n",
            "138\t1\t0.0\t1\t0 1\n",
            "139\t1\t0.0\t1\t0 1\n",
            "140\t4\t0.0\t1\t0 4\n",
            "141\t1\t0.0\t1\t0 1\n",
            "142\t1\t0.0\t1\t0 1\n",
            "143\t1\t0.0\t1\t0 1\n",
            "144\t1\t0.0\t1\t0 1\n",
            "145\t1\t0.0\t1\t0 1\n",
            "146\t3\t0.0\t1\t0 3\n",
            "147\t2\t0.0\t1\t0 2\n",
            "148\t2\t0.0\t1\t0 2\n",
            "149\t8\t0.0\t1\t0 8\n",
            "150\t120\t0.0\t1\t0 120\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_NW13_sub_1.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_NW13_sub_2.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_NW13_sub_2.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_NW13_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_NW13_sub_2.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_NW13_sub_2.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 55.007 s (55.007 µs/read; 1.09 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   369,861 (37.0%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   150,000,000 bp\n",
            "Quality-trimmed:              10,276,320 bp (6.9%)\n",
            "Total written (filtered):    138,025,015 bp (92.0%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 369861 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 32.4%\n",
            "  C: 25.0%\n",
            "  G: 18.6%\n",
            "  T: 20.1%\n",
            "  none/other: 3.9%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t237096\t250000.0\t0\t237096\n",
            "2\t69834\t62500.0\t0\t69834\n",
            "3\t19631\t15625.0\t0\t19631\n",
            "4\t3893\t3906.2\t0\t3893\n",
            "5\t1294\t976.6\t0\t1294\n",
            "6\t814\t244.1\t0\t814\n",
            "7\t438\t61.0\t0\t438\n",
            "8\t423\t15.3\t0\t423\n",
            "9\t437\t3.8\t0\t409 28\n",
            "10\t608\t1.0\t1\t361 247\n",
            "11\t1134\t0.2\t1\t396 738\n",
            "12\t967\t0.1\t1\t372 595\n",
            "13\t999\t0.0\t1\t380 619\n",
            "14\t1149\t0.0\t1\t520 629\n",
            "15\t898\t0.0\t1\t303 595\n",
            "16\t739\t0.0\t1\t351 388\n",
            "17\t773\t0.0\t1\t372 401\n",
            "18\t763\t0.0\t1\t320 443\n",
            "19\t913\t0.0\t1\t505 408\n",
            "20\t721\t0.0\t1\t675 46\n",
            "21\t713\t0.0\t1\t661 52\n",
            "22\t788\t0.0\t1\t685 103\n",
            "23\t920\t0.0\t1\t833 87\n",
            "24\t1128\t0.0\t1\t992 136\n",
            "25\t1152\t0.0\t1\t1075 77\n",
            "26\t1011\t0.0\t1\t943 68\n",
            "27\t871\t0.0\t1\t820 51\n",
            "28\t947\t0.0\t1\t874 73\n",
            "29\t901\t0.0\t1\t839 62\n",
            "30\t809\t0.0\t1\t748 61\n",
            "31\t635\t0.0\t1\t595 40\n",
            "32\t650\t0.0\t1\t611 39\n",
            "33\t532\t0.0\t1\t493 39\n",
            "34\t632\t0.0\t1\t589 43\n",
            "35\t611\t0.0\t1\t567 44\n",
            "36\t591\t0.0\t1\t553 38\n",
            "37\t604\t0.0\t1\t577 27\n",
            "38\t569\t0.0\t1\t551 18\n",
            "39\t456\t0.0\t1\t429 27\n",
            "40\t513\t0.0\t1\t486 27\n",
            "41\t513\t0.0\t1\t479 34\n",
            "42\t602\t0.0\t1\t578 24\n",
            "43\t519\t0.0\t1\t499 20\n",
            "44\t507\t0.0\t1\t494 13\n",
            "45\t464\t0.0\t1\t444 20\n",
            "46\t400\t0.0\t1\t377 23\n",
            "47\t380\t0.0\t1\t371 9\n",
            "48\t367\t0.0\t1\t351 16\n",
            "49\t349\t0.0\t1\t338 11\n",
            "50\t253\t0.0\t1\t246 7\n",
            "51\t321\t0.0\t1\t315 6\n",
            "52\t269\t0.0\t1\t261 8\n",
            "53\t183\t0.0\t1\t180 3\n",
            "54\t213\t0.0\t1\t209 4\n",
            "55\t197\t0.0\t1\t194 3\n",
            "56\t183\t0.0\t1\t181 2\n",
            "57\t206\t0.0\t1\t201 5\n",
            "58\t212\t0.0\t1\t207 5\n",
            "59\t208\t0.0\t1\t203 5\n",
            "60\t214\t0.0\t1\t213 1\n",
            "61\t290\t0.0\t1\t283 7\n",
            "62\t333\t0.0\t1\t321 12\n",
            "63\t690\t0.0\t1\t385 305\n",
            "64\t1766\t0.0\t1\t1589 177\n",
            "65\t559\t0.0\t1\t509 50\n",
            "66\t212\t0.0\t1\t194 18\n",
            "67\t96\t0.0\t1\t81 15\n",
            "68\t56\t0.0\t1\t46 10\n",
            "69\t41\t0.0\t1\t36 5\n",
            "70\t40\t0.0\t1\t36 4\n",
            "71\t32\t0.0\t1\t29 3\n",
            "72\t33\t0.0\t1\t29 4\n",
            "73\t24\t0.0\t1\t22 2\n",
            "74\t25\t0.0\t1\t19 6\n",
            "75\t15\t0.0\t1\t13 2\n",
            "76\t29\t0.0\t1\t23 6\n",
            "77\t19\t0.0\t1\t18 1\n",
            "78\t28\t0.0\t1\t22 6\n",
            "79\t12\t0.0\t1\t10 2\n",
            "80\t19\t0.0\t1\t15 4\n",
            "81\t14\t0.0\t1\t12 2\n",
            "82\t18\t0.0\t1\t16 2\n",
            "83\t18\t0.0\t1\t12 6\n",
            "84\t11\t0.0\t1\t9 2\n",
            "85\t16\t0.0\t1\t12 4\n",
            "86\t15\t0.0\t1\t12 3\n",
            "87\t9\t0.0\t1\t8 1\n",
            "88\t11\t0.0\t1\t9 2\n",
            "89\t8\t0.0\t1\t7 1\n",
            "90\t10\t0.0\t1\t8 2\n",
            "91\t16\t0.0\t1\t14 2\n",
            "92\t8\t0.0\t1\t5 3\n",
            "93\t8\t0.0\t1\t4 4\n",
            "94\t10\t0.0\t1\t9 1\n",
            "95\t8\t0.0\t1\t4 4\n",
            "96\t12\t0.0\t1\t10 2\n",
            "97\t5\t0.0\t1\t3 2\n",
            "98\t5\t0.0\t1\t4 1\n",
            "99\t5\t0.0\t1\t4 1\n",
            "100\t14\t0.0\t1\t11 3\n",
            "101\t10\t0.0\t1\t8 2\n",
            "102\t9\t0.0\t1\t7 2\n",
            "103\t6\t0.0\t1\t6\n",
            "104\t6\t0.0\t1\t3 3\n",
            "105\t3\t0.0\t1\t3\n",
            "106\t3\t0.0\t1\t2 1\n",
            "107\t4\t0.0\t1\t2 2\n",
            "108\t1\t0.0\t1\t1\n",
            "109\t1\t0.0\t1\t0 1\n",
            "110\t7\t0.0\t1\t4 3\n",
            "111\t6\t0.0\t1\t2 4\n",
            "112\t1\t0.0\t1\t1\n",
            "113\t5\t0.0\t1\t2 3\n",
            "114\t3\t0.0\t1\t2 1\n",
            "115\t3\t0.0\t1\t2 1\n",
            "116\t4\t0.0\t1\t3 1\n",
            "117\t1\t0.0\t1\t1\n",
            "118\t7\t0.0\t1\t4 3\n",
            "119\t3\t0.0\t1\t3\n",
            "120\t4\t0.0\t1\t0 4\n",
            "121\t1\t0.0\t1\t1\n",
            "122\t2\t0.0\t1\t1 1\n",
            "123\t3\t0.0\t1\t1 2\n",
            "124\t1\t0.0\t1\t0 1\n",
            "125\t3\t0.0\t1\t1 2\n",
            "126\t3\t0.0\t1\t1 2\n",
            "127\t3\t0.0\t1\t2 1\n",
            "128\t2\t0.0\t1\t1 1\n",
            "129\t4\t0.0\t1\t1 3\n",
            "131\t1\t0.0\t1\t1\n",
            "132\t1\t0.0\t1\t0 1\n",
            "133\t1\t0.0\t1\t1\n",
            "134\t1\t0.0\t1\t0 1\n",
            "135\t1\t0.0\t1\t0 1\n",
            "137\t2\t0.0\t1\t1 1\n",
            "138\t2\t0.0\t1\t0 2\n",
            "141\t2\t0.0\t1\t0 2\n",
            "148\t1\t0.0\t1\t0 1\n",
            "149\t3\t0.0\t1\t0 3\n",
            "150\t80\t0.0\t1\t0 80\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_NW13_sub_2.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Validate paired-end files BEN_NW13_sub_1_trimmed.fq.gz and BEN_NW13_sub_2_trimmed.fq.gz\n",
            "file_1: BEN_NW13_sub_1_trimmed.fq.gz, file_2: BEN_NW13_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            ">>>>> Now validing the length of the 2 paired-end infiles: BEN_NW13_sub_1_trimmed.fq.gz and BEN_NW13_sub_2_trimmed.fq.gz <<<<<\n",
            "Writing validated paired-end Read 1 reads to BEN_NW13_sub_1_val_1.fq.gz\n",
            "Writing validated paired-end Read 2 reads to BEN_NW13_sub_2_val_2.fq.gz\n",
            "\n",
            "Total number of sequences analysed: 1000000\n",
            "\n",
            "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 8123 (0.81%)\n",
            "\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_NW13_sub_1_val_1.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 5% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 10% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 15% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 20% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 25% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 30% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 35% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 40% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 45% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 50% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 55% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 60% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 65% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 70% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 75% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 80% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 85% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 90% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Approx 95% complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "Analysis complete for BEN_NW13_sub_1_val_1.fq.gz\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_NW13_sub_2_val_2.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 5% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 10% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 15% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 20% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 25% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 30% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 35% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 40% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 45% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 50% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 55% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 60% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 65% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 70% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 75% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 80% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 85% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 90% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Approx 95% complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Analysis complete for BEN_NW13_sub_2_val_2.fq.gz\n",
            "Deleting both intermediate output files BEN_NW13_sub_1_trimmed.fq.gz and BEN_NW13_sub_2_trimmed.fq.gz\n",
            "\n",
            "====================================================================================================\n",
            "\n",
            "Multicore support not enabled. Proceeding with single-core trimming.\n",
            "Path to Cutadapt set as: 'cutadapt' (default)\n",
            "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
            "Cutadapt version: 4.4\n",
            "single-core operation.\n",
            "igzip command line interface 2.31.1\n",
            "igzip detected. Using igzip for decompressing\n",
            "\n",
            "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
            "\n",
            "Output will be written into the directory: /content/trimmed_fastq_files/\n",
            "\n",
            "\n",
            "AUTO-DETECTING ADAPTER TYPE\n",
            "===========================\n",
            "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastq_files/BEN_SI18_sub_1.fq.gz <<)\n",
            "\n",
            "Found perfect matches for the following adapter sequences:\n",
            "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
            "Illumina\t22026\tAGATCGGAAGAGC\t1000000\t2.20\n",
            "Nextera\t14\tCTGTCTCTTATA\t1000000\t0.00\n",
            "smallRNA\t3\tTGGAATTCTCGG\t1000000\t0.00\n",
            "Using Illumina adapter for trimming (count: 22026). Second best hit was Nextera (count: 14)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_SI18_sub_1.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_SI18_sub_1.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_SI18_sub_1_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_SI18_sub_1.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_SI18_sub_1.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 57.661 s (57.661 µs/read; 1.04 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   368,958 (36.9%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   151,000,000 bp\n",
            "Quality-trimmed:                 633,875 bp (0.4%)\n",
            "Total written (filtered):    148,976,807 bp (98.7%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 368958 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 32.9%\n",
            "  C: 27.4%\n",
            "  G: 18.5%\n",
            "  T: 21.1%\n",
            "  none/other: 0.1%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t239031\t250000.0\t0\t239031\n",
            "2\t72499\t62500.0\t0\t72499\n",
            "3\t22041\t15625.0\t0\t22041\n",
            "4\t4660\t3906.2\t0\t4660\n",
            "5\t1744\t976.6\t0\t1744\n",
            "6\t976\t244.1\t0\t976\n",
            "7\t874\t61.0\t0\t874\n",
            "8\t860\t15.3\t0\t860\n",
            "9\t857\t3.8\t0\t838 19\n",
            "10\t889\t1.0\t1\t843 46\n",
            "11\t791\t0.2\t1\t756 35\n",
            "12\t744\t0.1\t1\t712 32\n",
            "13\t750\t0.0\t1\t721 29\n",
            "14\t745\t0.0\t1\t728 17\n",
            "15\t687\t0.0\t1\t669 18\n",
            "16\t726\t0.0\t1\t705 21\n",
            "17\t725\t0.0\t1\t698 27\n",
            "18\t714\t0.0\t1\t692 22\n",
            "19\t611\t0.0\t1\t587 24\n",
            "20\t665\t0.0\t1\t635 30\n",
            "21\t545\t0.0\t1\t531 14\n",
            "22\t576\t0.0\t1\t553 23\n",
            "23\t551\t0.0\t1\t536 15\n",
            "24\t576\t0.0\t1\t555 21\n",
            "25\t655\t0.0\t1\t619 36\n",
            "26\t391\t0.0\t1\t375 16\n",
            "27\t456\t0.0\t1\t436 20\n",
            "28\t517\t0.0\t1\t498 19\n",
            "29\t464\t0.0\t1\t454 10\n",
            "30\t555\t0.0\t1\t532 23\n",
            "31\t389\t0.0\t1\t383 6\n",
            "32\t435\t0.0\t1\t423 12\n",
            "33\t387\t0.0\t1\t376 11\n",
            "34\t397\t0.0\t1\t386 11\n",
            "35\t517\t0.0\t1\t495 22\n",
            "36\t260\t0.0\t1\t252 8\n",
            "37\t402\t0.0\t1\t396 6\n",
            "38\t361\t0.0\t1\t351 10\n",
            "39\t349\t0.0\t1\t339 10\n",
            "40\t320\t0.0\t1\t313 7\n",
            "41\t344\t0.0\t1\t335 9\n",
            "42\t283\t0.0\t1\t271 12\n",
            "43\t358\t0.0\t1\t346 12\n",
            "44\t241\t0.0\t1\t238 3\n",
            "45\t486\t0.0\t1\t479 7\n",
            "46\t75\t0.0\t1\t74 1\n",
            "47\t204\t0.0\t1\t203 1\n",
            "48\t268\t0.0\t1\t259 9\n",
            "49\t211\t0.0\t1\t208 3\n",
            "50\t246\t0.0\t1\t239 7\n",
            "51\t200\t0.0\t1\t192 8\n",
            "52\t217\t0.0\t1\t210 7\n",
            "53\t229\t0.0\t1\t221 8\n",
            "54\t219\t0.0\t1\t211 8\n",
            "55\t243\t0.0\t1\t238 5\n",
            "56\t132\t0.0\t1\t131 1\n",
            "57\t175\t0.0\t1\t173 2\n",
            "58\t208\t0.0\t1\t201 7\n",
            "59\t148\t0.0\t1\t146 2\n",
            "60\t170\t0.0\t1\t169 1\n",
            "61\t127\t0.0\t1\t123 4\n",
            "62\t144\t0.0\t1\t139 5\n",
            "63\t170\t0.0\t1\t162 8\n",
            "64\t126\t0.0\t1\t123 3\n",
            "65\t105\t0.0\t1\t105\n",
            "66\t112\t0.0\t1\t111 1\n",
            "67\t121\t0.0\t1\t118 3\n",
            "68\t115\t0.0\t1\t109 6\n",
            "69\t145\t0.0\t1\t136 9\n",
            "70\t135\t0.0\t1\t126 9\n",
            "71\t307\t0.0\t1\t145 162\n",
            "72\t1067\t0.0\t1\t1004 63\n",
            "73\t233\t0.0\t1\t219 14\n",
            "74\t78\t0.0\t1\t63 15\n",
            "75\t35\t0.0\t1\t30 5\n",
            "76\t38\t0.0\t1\t27 11\n",
            "77\t19\t0.0\t1\t17 2\n",
            "78\t21\t0.0\t1\t18 3\n",
            "79\t20\t0.0\t1\t18 2\n",
            "80\t22\t0.0\t1\t18 4\n",
            "81\t21\t0.0\t1\t17 4\n",
            "82\t29\t0.0\t1\t23 6\n",
            "83\t13\t0.0\t1\t12 1\n",
            "84\t14\t0.0\t1\t11 3\n",
            "85\t16\t0.0\t1\t15 1\n",
            "86\t21\t0.0\t1\t19 2\n",
            "87\t8\t0.0\t1\t6 2\n",
            "88\t17\t0.0\t1\t16 1\n",
            "89\t13\t0.0\t1\t10 3\n",
            "90\t12\t0.0\t1\t11 1\n",
            "91\t11\t0.0\t1\t9 2\n",
            "92\t12\t0.0\t1\t11 1\n",
            "93\t13\t0.0\t1\t11 2\n",
            "94\t3\t0.0\t1\t3\n",
            "95\t16\t0.0\t1\t11 5\n",
            "96\t11\t0.0\t1\t9 2\n",
            "97\t11\t0.0\t1\t10 1\n",
            "98\t7\t0.0\t1\t5 2\n",
            "99\t10\t0.0\t1\t8 2\n",
            "100\t6\t0.0\t1\t5 1\n",
            "101\t3\t0.0\t1\t3\n",
            "102\t7\t0.0\t1\t7\n",
            "103\t4\t0.0\t1\t4\n",
            "104\t5\t0.0\t1\t3 2\n",
            "105\t7\t0.0\t1\t2 5\n",
            "106\t5\t0.0\t1\t3 2\n",
            "107\t10\t0.0\t1\t8 2\n",
            "108\t5\t0.0\t1\t3 2\n",
            "109\t4\t0.0\t1\t4\n",
            "110\t4\t0.0\t1\t3 1\n",
            "111\t4\t0.0\t1\t4\n",
            "112\t7\t0.0\t1\t4 3\n",
            "113\t6\t0.0\t1\t4 2\n",
            "114\t3\t0.0\t1\t3\n",
            "115\t3\t0.0\t1\t3\n",
            "116\t4\t0.0\t1\t3 1\n",
            "117\t1\t0.0\t1\t1\n",
            "118\t3\t0.0\t1\t0 3\n",
            "119\t2\t0.0\t1\t1 1\n",
            "120\t3\t0.0\t1\t2 1\n",
            "121\t2\t0.0\t1\t2\n",
            "122\t6\t0.0\t1\t5 1\n",
            "123\t1\t0.0\t1\t1\n",
            "124\t2\t0.0\t1\t0 2\n",
            "125\t1\t0.0\t1\t0 1\n",
            "127\t3\t0.0\t1\t1 2\n",
            "128\t2\t0.0\t1\t0 2\n",
            "129\t1\t0.0\t1\t0 1\n",
            "130\t2\t0.0\t1\t0 2\n",
            "132\t1\t0.0\t1\t0 1\n",
            "133\t2\t0.0\t1\t1 1\n",
            "134\t1\t0.0\t1\t0 1\n",
            "135\t1\t0.0\t1\t0 1\n",
            "136\t3\t0.0\t1\t1 2\n",
            "137\t3\t0.0\t1\t0 3\n",
            "138\t1\t0.0\t1\t0 1\n",
            "140\t3\t0.0\t1\t0 3\n",
            "141\t2\t0.0\t1\t1 1\n",
            "142\t1\t0.0\t1\t0 1\n",
            "143\t1\t0.0\t1\t0 1\n",
            "144\t2\t0.0\t1\t0 2\n",
            "146\t2\t0.0\t1\t1 1\n",
            "147\t2\t0.0\t1\t0 2\n",
            "148\t3\t0.0\t1\t0 3\n",
            "149\t2\t0.0\t1\t1 1\n",
            "150\t8\t0.0\t1\t0 8\n",
            "151\t57\t0.0\t1\t2 55\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_SI18_sub_1.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_SI18_sub_2.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_SI18_sub_2.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_SI18_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_SI18_sub_2.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_SI18_sub_2.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 63.118 s (63.118 µs/read; 0.95 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   313,039 (31.3%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   151,000,000 bp\n",
            "Quality-trimmed:               4,288,982 bp (2.8%)\n",
            "Total written (filtered):    145,393,149 bp (96.3%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 313039 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 34.4%\n",
            "  C: 26.7%\n",
            "  G: 17.7%\n",
            "  T: 21.0%\n",
            "  none/other: 0.1%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t182649\t250000.0\t0\t182649\n",
            "2\t81188\t62500.0\t0\t81188\n",
            "3\t13945\t15625.0\t0\t13945\n",
            "4\t4363\t3906.2\t0\t4363\n",
            "5\t1714\t976.6\t0\t1714\n",
            "6\t1006\t244.1\t0\t1006\n",
            "7\t1202\t61.0\t0\t1202\n",
            "8\t695\t15.3\t0\t695\n",
            "9\t710\t3.8\t0\t687 23\n",
            "10\t1240\t1.0\t1\t1177 63\n",
            "11\t358\t0.2\t1\t329 29\n",
            "12\t866\t0.1\t1\t813 53\n",
            "13\t598\t0.0\t1\t574 24\n",
            "14\t1259\t0.0\t1\t1188 71\n",
            "15\t369\t0.0\t1\t348 21\n",
            "16\t688\t0.0\t1\t655 33\n",
            "17\t1119\t0.0\t1\t1078 41\n",
            "18\t185\t0.0\t1\t173 12\n",
            "19\t904\t0.0\t1\t849 55\n",
            "20\t591\t0.0\t1\t561 30\n",
            "21\t272\t0.0\t1\t254 18\n",
            "22\t545\t0.0\t1\t516 29\n",
            "23\t586\t0.0\t1\t555 31\n",
            "24\t1066\t0.0\t1\t1000 66\n",
            "25\t261\t0.0\t1\t242 19\n",
            "26\t466\t0.0\t1\t441 25\n",
            "27\t339\t0.0\t1\t325 14\n",
            "28\t762\t0.0\t1\t743 19\n",
            "29\t180\t0.0\t1\t169 11\n",
            "30\t821\t0.0\t1\t784 37\n",
            "31\t132\t0.0\t1\t117 15\n",
            "32\t502\t0.0\t1\t482 20\n",
            "33\t441\t0.0\t1\t415 26\n",
            "34\t272\t0.0\t1\t266 6\n",
            "35\t517\t0.0\t1\t490 27\n",
            "36\t257\t0.0\t1\t237 20\n",
            "37\t405\t0.0\t1\t387 18\n",
            "38\t327\t0.0\t1\t311 16\n",
            "39\t349\t0.0\t1\t324 25\n",
            "40\t303\t0.0\t1\t289 14\n",
            "41\t349\t0.0\t1\t329 20\n",
            "42\t455\t0.0\t1\t438 17\n",
            "43\t126\t0.0\t1\t119 7\n",
            "44\t315\t0.0\t1\t312 3\n",
            "45\t421\t0.0\t1\t399 22\n",
            "46\t198\t0.0\t1\t190 8\n",
            "47\t169\t0.0\t1\t159 10\n",
            "48\t344\t0.0\t1\t328 16\n",
            "49\t195\t0.0\t1\t182 13\n",
            "50\t228\t0.0\t1\t216 12\n",
            "51\t510\t0.0\t1\t491 19\n",
            "52\t103\t0.0\t1\t94 9\n",
            "53\t134\t0.0\t1\t131 3\n",
            "54\t176\t0.0\t1\t168 8\n",
            "55\t238\t0.0\t1\t227 11\n",
            "56\t148\t0.0\t1\t143 5\n",
            "57\t174\t0.0\t1\t167 7\n",
            "58\t224\t0.0\t1\t220 4\n",
            "59\t173\t0.0\t1\t164 9\n",
            "60\t159\t0.0\t1\t154 5\n",
            "61\t178\t0.0\t1\t174 4\n",
            "62\t181\t0.0\t1\t173 8\n",
            "63\t414\t0.0\t1\t219 195\n",
            "64\t1759\t0.0\t1\t1691 68\n",
            "65\t377\t0.0\t1\t355 22\n",
            "66\t106\t0.0\t1\t87 19\n",
            "67\t54\t0.0\t1\t47 7\n",
            "68\t34\t0.0\t1\t28 6\n",
            "69\t28\t0.0\t1\t25 3\n",
            "70\t18\t0.0\t1\t15 3\n",
            "71\t17\t0.0\t1\t14 3\n",
            "72\t20\t0.0\t1\t15 5\n",
            "73\t13\t0.0\t1\t9 4\n",
            "74\t33\t0.0\t1\t28 5\n",
            "75\t25\t0.0\t1\t21 4\n",
            "76\t28\t0.0\t1\t23 5\n",
            "77\t20\t0.0\t1\t16 4\n",
            "78\t9\t0.0\t1\t8 1\n",
            "79\t14\t0.0\t1\t8 6\n",
            "80\t10\t0.0\t1\t9 1\n",
            "81\t11\t0.0\t1\t8 3\n",
            "82\t15\t0.0\t1\t15\n",
            "83\t11\t0.0\t1\t10 1\n",
            "84\t16\t0.0\t1\t16\n",
            "85\t3\t0.0\t1\t3\n",
            "86\t9\t0.0\t1\t6 3\n",
            "87\t10\t0.0\t1\t8 2\n",
            "88\t8\t0.0\t1\t6 2\n",
            "89\t9\t0.0\t1\t7 2\n",
            "90\t11\t0.0\t1\t8 3\n",
            "91\t9\t0.0\t1\t4 5\n",
            "92\t11\t0.0\t1\t7 4\n",
            "93\t6\t0.0\t1\t5 1\n",
            "94\t2\t0.0\t1\t1 1\n",
            "95\t7\t0.0\t1\t5 2\n",
            "96\t6\t0.0\t1\t3 3\n",
            "97\t2\t0.0\t1\t2\n",
            "98\t7\t0.0\t1\t5 2\n",
            "99\t10\t0.0\t1\t7 3\n",
            "100\t7\t0.0\t1\t6 1\n",
            "101\t6\t0.0\t1\t5 1\n",
            "102\t6\t0.0\t1\t4 2\n",
            "103\t10\t0.0\t1\t7 3\n",
            "104\t11\t0.0\t1\t4 7\n",
            "105\t5\t0.0\t1\t5\n",
            "106\t10\t0.0\t1\t7 3\n",
            "107\t6\t0.0\t1\t5 1\n",
            "108\t9\t0.0\t1\t5 4\n",
            "109\t5\t0.0\t1\t4 1\n",
            "110\t7\t0.0\t1\t5 2\n",
            "111\t7\t0.0\t1\t5 2\n",
            "112\t8\t0.0\t1\t5 3\n",
            "113\t6\t0.0\t1\t5 1\n",
            "114\t7\t0.0\t1\t6 1\n",
            "115\t6\t0.0\t1\t6\n",
            "116\t7\t0.0\t1\t2 5\n",
            "117\t2\t0.0\t1\t1 1\n",
            "118\t4\t0.0\t1\t2 2\n",
            "119\t3\t0.0\t1\t3\n",
            "120\t5\t0.0\t1\t4 1\n",
            "121\t7\t0.0\t1\t4 3\n",
            "122\t4\t0.0\t1\t3 1\n",
            "123\t3\t0.0\t1\t0 3\n",
            "124\t4\t0.0\t1\t1 3\n",
            "125\t3\t0.0\t1\t1 2\n",
            "126\t1\t0.0\t1\t1\n",
            "127\t3\t0.0\t1\t1 2\n",
            "128\t4\t0.0\t1\t2 2\n",
            "129\t4\t0.0\t1\t2 2\n",
            "130\t2\t0.0\t1\t1 1\n",
            "131\t3\t0.0\t1\t2 1\n",
            "132\t4\t0.0\t1\t2 2\n",
            "134\t2\t0.0\t1\t0 2\n",
            "135\t3\t0.0\t1\t1 2\n",
            "136\t2\t0.0\t1\t2\n",
            "137\t1\t0.0\t1\t1\n",
            "138\t2\t0.0\t1\t1 1\n",
            "139\t1\t0.0\t1\t1\n",
            "140\t3\t0.0\t1\t0 3\n",
            "141\t4\t0.0\t1\t1 3\n",
            "143\t1\t0.0\t1\t0 1\n",
            "145\t1\t0.0\t1\t0 1\n",
            "146\t2\t0.0\t1\t0 2\n",
            "147\t1\t0.0\t1\t1\n",
            "148\t1\t0.0\t1\t0 1\n",
            "149\t3\t0.0\t1\t0 3\n",
            "150\t7\t0.0\t1\t0 7\n",
            "151\t34\t0.0\t1\t0 34\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_SI18_sub_2.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Validate paired-end files BEN_SI18_sub_1_trimmed.fq.gz and BEN_SI18_sub_2_trimmed.fq.gz\n",
            "file_1: BEN_SI18_sub_1_trimmed.fq.gz, file_2: BEN_SI18_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            ">>>>> Now validing the length of the 2 paired-end infiles: BEN_SI18_sub_1_trimmed.fq.gz and BEN_SI18_sub_2_trimmed.fq.gz <<<<<\n",
            "Writing validated paired-end Read 1 reads to BEN_SI18_sub_1_val_1.fq.gz\n",
            "Writing validated paired-end Read 2 reads to BEN_SI18_sub_2_val_2.fq.gz\n",
            "\n",
            "Total number of sequences analysed: 1000000\n",
            "\n",
            "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 4246 (0.42%)\n",
            "\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_SI18_sub_1_val_1.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 5% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 10% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 15% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 20% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 25% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 30% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 35% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 40% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 45% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 50% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 55% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 60% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 65% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 70% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 75% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 80% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 85% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 90% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Approx 95% complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "Analysis complete for BEN_SI18_sub_1_val_1.fq.gz\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_SI18_sub_2_val_2.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 5% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 10% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 15% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 20% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 25% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 30% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 35% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 40% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 45% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 50% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 55% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 60% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 65% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 70% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 75% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 80% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 85% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 90% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Approx 95% complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Analysis complete for BEN_SI18_sub_2_val_2.fq.gz\n",
            "Deleting both intermediate output files BEN_SI18_sub_1_trimmed.fq.gz and BEN_SI18_sub_2_trimmed.fq.gz\n",
            "\n",
            "====================================================================================================\n",
            "\n",
            "Multicore support not enabled. Proceeding with single-core trimming.\n",
            "Path to Cutadapt set as: 'cutadapt' (default)\n",
            "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
            "Cutadapt version: 4.4\n",
            "single-core operation.\n",
            "igzip command line interface 2.31.1\n",
            "igzip detected. Using igzip for decompressing\n",
            "\n",
            "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
            "\n",
            "Output will be written into the directory: /content/trimmed_fastq_files/\n",
            "\n",
            "\n",
            "AUTO-DETECTING ADAPTER TYPE\n",
            "===========================\n",
            "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastq_files/BEN_SI19_sub_1.fq.gz <<)\n",
            "\n",
            "Found perfect matches for the following adapter sequences:\n",
            "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
            "Illumina\t26786\tAGATCGGAAGAGC\t1000000\t2.68\n",
            "Nextera\t6\tCTGTCTCTTATA\t1000000\t0.00\n",
            "smallRNA\t3\tTGGAATTCTCGG\t1000000\t0.00\n",
            "Using Illumina adapter for trimming (count: 26786). Second best hit was Nextera (count: 6)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_SI19_sub_1.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_SI19_sub_1.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_SI19_sub_1_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_SI19_sub_1.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_SI19_sub_1.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 57.878 s (57.878 µs/read; 1.04 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   374,839 (37.5%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   151,000,000 bp\n",
            "Quality-trimmed:                 616,107 bp (0.4%)\n",
            "Total written (filtered):    148,842,332 bp (98.6%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 374839 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 32.7%\n",
            "  C: 27.3%\n",
            "  G: 18.4%\n",
            "  T: 21.5%\n",
            "  none/other: 0.1%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t237004\t250000.0\t0\t237004\n",
            "2\t72326\t62500.0\t0\t72326\n",
            "3\t22014\t15625.0\t0\t22014\n",
            "4\t5079\t3906.2\t0\t5079\n",
            "5\t2144\t976.6\t0\t2144\n",
            "6\t1348\t244.1\t0\t1348\n",
            "7\t1331\t61.0\t0\t1331\n",
            "8\t1145\t15.3\t0\t1145\n",
            "9\t1294\t3.8\t0\t1275 19\n",
            "10\t1225\t1.0\t1\t1177 48\n",
            "11\t1041\t0.2\t1\t999 42\n",
            "12\t1087\t0.1\t1\t1058 29\n",
            "13\t983\t0.0\t1\t956 27\n",
            "14\t976\t0.0\t1\t945 31\n",
            "15\t931\t0.0\t1\t901 30\n",
            "16\t925\t0.0\t1\t893 32\n",
            "17\t907\t0.0\t1\t870 37\n",
            "18\t1048\t0.0\t1\t1012 36\n",
            "19\t699\t0.0\t1\t677 22\n",
            "20\t841\t0.0\t1\t813 28\n",
            "21\t691\t0.0\t1\t670 21\n",
            "22\t766\t0.0\t1\t745 21\n",
            "23\t753\t0.0\t1\t729 24\n",
            "24\t748\t0.0\t1\t717 31\n",
            "25\t738\t0.0\t1\t710 28\n",
            "26\t596\t0.0\t1\t574 22\n",
            "27\t637\t0.0\t1\t613 24\n",
            "28\t690\t0.0\t1\t670 20\n",
            "29\t645\t0.0\t1\t628 17\n",
            "30\t661\t0.0\t1\t648 13\n",
            "31\t485\t0.0\t1\t480 5\n",
            "32\t483\t0.0\t1\t464 19\n",
            "33\t488\t0.0\t1\t471 17\n",
            "34\t470\t0.0\t1\t455 15\n",
            "35\t507\t0.0\t1\t481 26\n",
            "36\t483\t0.0\t1\t465 18\n",
            "37\t387\t0.0\t1\t379 8\n",
            "38\t431\t0.0\t1\t420 11\n",
            "39\t401\t0.0\t1\t390 11\n",
            "40\t415\t0.0\t1\t404 11\n",
            "41\t398\t0.0\t1\t389 9\n",
            "42\t385\t0.0\t1\t372 13\n",
            "43\t475\t0.0\t1\t463 12\n",
            "44\t264\t0.0\t1\t257 7\n",
            "45\t596\t0.0\t1\t588 8\n",
            "46\t73\t0.0\t1\t69 4\n",
            "47\t250\t0.0\t1\t242 8\n",
            "48\t287\t0.0\t1\t281 6\n",
            "49\t265\t0.0\t1\t253 12\n",
            "50\t266\t0.0\t1\t259 7\n",
            "51\t226\t0.0\t1\t220 6\n",
            "52\t240\t0.0\t1\t234 6\n",
            "53\t233\t0.0\t1\t228 5\n",
            "54\t229\t0.0\t1\t225 4\n",
            "55\t279\t0.0\t1\t269 10\n",
            "56\t150\t0.0\t1\t147 3\n",
            "57\t192\t0.0\t1\t189 3\n",
            "58\t175\t0.0\t1\t172 3\n",
            "59\t167\t0.0\t1\t158 9\n",
            "60\t166\t0.0\t1\t162 4\n",
            "61\t147\t0.0\t1\t140 7\n",
            "62\t138\t0.0\t1\t131 7\n",
            "63\t187\t0.0\t1\t181 6\n",
            "64\t139\t0.0\t1\t137 2\n",
            "65\t139\t0.0\t1\t131 8\n",
            "66\t153\t0.0\t1\t150 3\n",
            "67\t102\t0.0\t1\t101 1\n",
            "68\t104\t0.0\t1\t103 1\n",
            "69\t128\t0.0\t1\t126 2\n",
            "70\t133\t0.0\t1\t125 8\n",
            "71\t289\t0.0\t1\t158 131\n",
            "72\t1088\t0.0\t1\t1030 58\n",
            "73\t305\t0.0\t1\t284 21\n",
            "74\t96\t0.0\t1\t85 11\n",
            "75\t33\t0.0\t1\t30 3\n",
            "76\t29\t0.0\t1\t23 6\n",
            "77\t15\t0.0\t1\t13 2\n",
            "78\t18\t0.0\t1\t16 2\n",
            "79\t15\t0.0\t1\t13 2\n",
            "80\t19\t0.0\t1\t17 2\n",
            "81\t19\t0.0\t1\t14 5\n",
            "82\t16\t0.0\t1\t12 4\n",
            "83\t25\t0.0\t1\t23 2\n",
            "84\t14\t0.0\t1\t14\n",
            "85\t11\t0.0\t1\t9 2\n",
            "86\t16\t0.0\t1\t16\n",
            "87\t11\t0.0\t1\t9 2\n",
            "88\t11\t0.0\t1\t9 2\n",
            "89\t10\t0.0\t1\t7 3\n",
            "90\t14\t0.0\t1\t11 3\n",
            "91\t7\t0.0\t1\t7\n",
            "92\t15\t0.0\t1\t14 1\n",
            "93\t10\t0.0\t1\t10\n",
            "94\t7\t0.0\t1\t6 1\n",
            "95\t11\t0.0\t1\t11\n",
            "96\t10\t0.0\t1\t9 1\n",
            "97\t9\t0.0\t1\t9\n",
            "98\t6\t0.0\t1\t5 1\n",
            "99\t8\t0.0\t1\t6 2\n",
            "100\t7\t0.0\t1\t5 2\n",
            "101\t6\t0.0\t1\t5 1\n",
            "102\t7\t0.0\t1\t6 1\n",
            "103\t8\t0.0\t1\t6 2\n",
            "104\t5\t0.0\t1\t4 1\n",
            "105\t8\t0.0\t1\t7 1\n",
            "106\t2\t0.0\t1\t2\n",
            "107\t2\t0.0\t1\t2\n",
            "108\t8\t0.0\t1\t8\n",
            "109\t8\t0.0\t1\t7 1\n",
            "110\t4\t0.0\t1\t2 2\n",
            "111\t3\t0.0\t1\t2 1\n",
            "112\t6\t0.0\t1\t3 3\n",
            "113\t2\t0.0\t1\t2\n",
            "114\t3\t0.0\t1\t3\n",
            "115\t3\t0.0\t1\t2 1\n",
            "116\t3\t0.0\t1\t1 2\n",
            "117\t6\t0.0\t1\t4 2\n",
            "118\t3\t0.0\t1\t1 2\n",
            "119\t4\t0.0\t1\t3 1\n",
            "122\t1\t0.0\t1\t1\n",
            "123\t5\t0.0\t1\t3 2\n",
            "124\t1\t0.0\t1\t0 1\n",
            "125\t6\t0.0\t1\t4 2\n",
            "126\t1\t0.0\t1\t1\n",
            "127\t2\t0.0\t1\t0 2\n",
            "128\t1\t0.0\t1\t1\n",
            "130\t1\t0.0\t1\t1\n",
            "131\t1\t0.0\t1\t0 1\n",
            "134\t2\t0.0\t1\t1 1\n",
            "135\t1\t0.0\t1\t0 1\n",
            "137\t3\t0.0\t1\t1 2\n",
            "138\t1\t0.0\t1\t0 1\n",
            "139\t2\t0.0\t1\t0 2\n",
            "140\t1\t0.0\t1\t1\n",
            "141\t4\t0.0\t1\t2 2\n",
            "142\t1\t0.0\t1\t0 1\n",
            "143\t1\t0.0\t1\t0 1\n",
            "146\t2\t0.0\t1\t0 2\n",
            "147\t5\t0.0\t1\t1 4\n",
            "148\t1\t0.0\t1\t0 1\n",
            "149\t5\t0.0\t1\t0 5\n",
            "150\t9\t0.0\t1\t0 9\n",
            "151\t38\t0.0\t1\t0 38\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_SI19_sub_1.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_SI19_sub_2.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_SI19_sub_2.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_SI19_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_SI19_sub_2.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_SI19_sub_2.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 63.148 s (63.148 µs/read; 0.95 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   321,259 (32.1%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   151,000,000 bp\n",
            "Quality-trimmed:               3,492,656 bp (2.3%)\n",
            "Total written (filtered):    146,045,827 bp (96.7%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 321259 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 34.0%\n",
            "  C: 26.8%\n",
            "  G: 17.6%\n",
            "  T: 21.5%\n",
            "  none/other: 0.1%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t182218\t250000.0\t0\t182218\n",
            "2\t81710\t62500.0\t0\t81710\n",
            "3\t14091\t15625.0\t0\t14091\n",
            "4\t4927\t3906.2\t0\t4927\n",
            "5\t2171\t976.6\t0\t2171\n",
            "6\t1403\t244.1\t0\t1403\n",
            "7\t1719\t61.0\t0\t1719\n",
            "8\t992\t15.3\t0\t992\n",
            "9\t1057\t3.8\t0\t1037 20\n",
            "10\t1697\t1.0\t1\t1614 83\n",
            "11\t489\t0.2\t1\t452 37\n",
            "12\t1247\t0.1\t1\t1156 91\n",
            "13\t803\t0.0\t1\t757 46\n",
            "14\t1665\t0.0\t1\t1580 85\n",
            "15\t450\t0.0\t1\t420 30\n",
            "16\t892\t0.0\t1\t856 36\n",
            "17\t1411\t0.0\t1\t1338 73\n",
            "18\t240\t0.0\t1\t219 21\n",
            "19\t1162\t0.0\t1\t1123 39\n",
            "20\t746\t0.0\t1\t714 32\n",
            "21\t323\t0.0\t1\t302 21\n",
            "22\t741\t0.0\t1\t700 41\n",
            "23\t787\t0.0\t1\t752 35\n",
            "24\t1370\t0.0\t1\t1292 78\n",
            "25\t318\t0.0\t1\t301 17\n",
            "26\t610\t0.0\t1\t584 26\n",
            "27\t468\t0.0\t1\t455 13\n",
            "28\t998\t0.0\t1\t965 33\n",
            "29\t216\t0.0\t1\t201 15\n",
            "30\t995\t0.0\t1\t957 38\n",
            "31\t145\t0.0\t1\t127 18\n",
            "32\t588\t0.0\t1\t569 19\n",
            "33\t536\t0.0\t1\t516 20\n",
            "34\t323\t0.0\t1\t308 15\n",
            "35\t667\t0.0\t1\t639 28\n",
            "36\t283\t0.0\t1\t269 14\n",
            "37\t426\t0.0\t1\t415 11\n",
            "38\t406\t0.0\t1\t385 21\n",
            "39\t395\t0.0\t1\t373 22\n",
            "40\t369\t0.0\t1\t354 15\n",
            "41\t398\t0.0\t1\t382 16\n",
            "42\t570\t0.0\t1\t541 29\n",
            "43\t174\t0.0\t1\t169 5\n",
            "44\t374\t0.0\t1\t359 15\n",
            "45\t510\t0.0\t1\t488 22\n",
            "46\t211\t0.0\t1\t198 13\n",
            "47\t213\t0.0\t1\t206 7\n",
            "48\t380\t0.0\t1\t370 10\n",
            "49\t218\t0.0\t1\t207 11\n",
            "50\t226\t0.0\t1\t211 15\n",
            "51\t518\t0.0\t1\t493 25\n",
            "52\t102\t0.0\t1\t94 8\n",
            "53\t159\t0.0\t1\t154 5\n",
            "54\t172\t0.0\t1\t167 5\n",
            "55\t273\t0.0\t1\t266 7\n",
            "56\t156\t0.0\t1\t153 3\n",
            "57\t185\t0.0\t1\t178 7\n",
            "58\t228\t0.0\t1\t217 11\n",
            "59\t185\t0.0\t1\t176 9\n",
            "60\t184\t0.0\t1\t178 6\n",
            "61\t181\t0.0\t1\t177 4\n",
            "62\t173\t0.0\t1\t162 11\n",
            "63\t370\t0.0\t1\t210 160\n",
            "64\t1837\t0.0\t1\t1763 74\n",
            "65\t374\t0.0\t1\t354 20\n",
            "66\t104\t0.0\t1\t91 13\n",
            "67\t62\t0.0\t1\t53 9\n",
            "68\t38\t0.0\t1\t31 7\n",
            "69\t31\t0.0\t1\t28 3\n",
            "70\t21\t0.0\t1\t18 3\n",
            "71\t37\t0.0\t1\t31 6\n",
            "72\t24\t0.0\t1\t21 3\n",
            "73\t16\t0.0\t1\t14 2\n",
            "74\t26\t0.0\t1\t24 2\n",
            "75\t12\t0.0\t1\t11 1\n",
            "76\t27\t0.0\t1\t23 4\n",
            "77\t11\t0.0\t1\t8 3\n",
            "78\t13\t0.0\t1\t12 1\n",
            "79\t10\t0.0\t1\t7 3\n",
            "80\t14\t0.0\t1\t8 6\n",
            "81\t17\t0.0\t1\t14 3\n",
            "82\t15\t0.0\t1\t12 3\n",
            "83\t10\t0.0\t1\t8 2\n",
            "84\t12\t0.0\t1\t11 1\n",
            "85\t9\t0.0\t1\t9\n",
            "86\t18\t0.0\t1\t13 5\n",
            "87\t7\t0.0\t1\t6 1\n",
            "88\t10\t0.0\t1\t8 2\n",
            "89\t11\t0.0\t1\t10 1\n",
            "90\t4\t0.0\t1\t4\n",
            "91\t5\t0.0\t1\t3 2\n",
            "92\t5\t0.0\t1\t2 3\n",
            "93\t4\t0.0\t1\t4\n",
            "94\t7\t0.0\t1\t5 2\n",
            "95\t6\t0.0\t1\t5 1\n",
            "96\t7\t0.0\t1\t7\n",
            "97\t10\t0.0\t1\t8 2\n",
            "98\t3\t0.0\t1\t2 1\n",
            "99\t7\t0.0\t1\t6 1\n",
            "100\t3\t0.0\t1\t2 1\n",
            "101\t6\t0.0\t1\t4 2\n",
            "102\t12\t0.0\t1\t8 4\n",
            "103\t9\t0.0\t1\t7 2\n",
            "104\t7\t0.0\t1\t6 1\n",
            "105\t4\t0.0\t1\t2 2\n",
            "106\t11\t0.0\t1\t8 3\n",
            "107\t7\t0.0\t1\t2 5\n",
            "108\t4\t0.0\t1\t2 2\n",
            "109\t5\t0.0\t1\t2 3\n",
            "111\t3\t0.0\t1\t1 2\n",
            "112\t5\t0.0\t1\t4 1\n",
            "113\t3\t0.0\t1\t3\n",
            "114\t4\t0.0\t1\t2 2\n",
            "115\t7\t0.0\t1\t5 2\n",
            "116\t6\t0.0\t1\t3 3\n",
            "117\t4\t0.0\t1\t2 2\n",
            "118\t3\t0.0\t1\t3\n",
            "119\t4\t0.0\t1\t1 3\n",
            "120\t3\t0.0\t1\t2 1\n",
            "121\t4\t0.0\t1\t1 3\n",
            "122\t3\t0.0\t1\t2 1\n",
            "123\t3\t0.0\t1\t2 1\n",
            "124\t2\t0.0\t1\t1 1\n",
            "125\t3\t0.0\t1\t2 1\n",
            "126\t1\t0.0\t1\t0 1\n",
            "127\t5\t0.0\t1\t2 3\n",
            "128\t4\t0.0\t1\t3 1\n",
            "129\t5\t0.0\t1\t3 2\n",
            "130\t3\t0.0\t1\t1 2\n",
            "131\t2\t0.0\t1\t2\n",
            "132\t8\t0.0\t1\t3 5\n",
            "133\t3\t0.0\t1\t2 1\n",
            "134\t4\t0.0\t1\t1 3\n",
            "135\t3\t0.0\t1\t1 2\n",
            "136\t2\t0.0\t1\t0 2\n",
            "137\t5\t0.0\t1\t1 4\n",
            "138\t3\t0.0\t1\t2 1\n",
            "139\t3\t0.0\t1\t1 2\n",
            "140\t3\t0.0\t1\t1 2\n",
            "141\t2\t0.0\t1\t1 1\n",
            "144\t1\t0.0\t1\t0 1\n",
            "145\t1\t0.0\t1\t0 1\n",
            "146\t2\t0.0\t1\t0 2\n",
            "148\t1\t0.0\t1\t0 1\n",
            "149\t1\t0.0\t1\t0 1\n",
            "150\t2\t0.0\t1\t0 2\n",
            "151\t32\t0.0\t1\t2 30\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_SI19_sub_2.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Validate paired-end files BEN_SI19_sub_1_trimmed.fq.gz and BEN_SI19_sub_2_trimmed.fq.gz\n",
            "file_1: BEN_SI19_sub_1_trimmed.fq.gz, file_2: BEN_SI19_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            ">>>>> Now validing the length of the 2 paired-end infiles: BEN_SI19_sub_1_trimmed.fq.gz and BEN_SI19_sub_2_trimmed.fq.gz <<<<<\n",
            "Writing validated paired-end Read 1 reads to BEN_SI19_sub_1_val_1.fq.gz\n",
            "Writing validated paired-end Read 2 reads to BEN_SI19_sub_2_val_2.fq.gz\n",
            "\n",
            "Total number of sequences analysed: 1000000\n",
            "\n",
            "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 3107 (0.31%)\n",
            "\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_SI19_sub_1_val_1.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 5% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 10% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 15% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 20% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 25% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 30% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 35% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 40% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 45% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 50% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 55% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 60% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 65% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 70% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 75% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 80% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 85% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 90% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Approx 95% complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "Analysis complete for BEN_SI19_sub_1_val_1.fq.gz\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_SI19_sub_2_val_2.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 5% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 10% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 15% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 20% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 25% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 30% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 35% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 40% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 45% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 50% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 55% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 60% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 65% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 70% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 75% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 80% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 85% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 90% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Approx 95% complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Analysis complete for BEN_SI19_sub_2_val_2.fq.gz\n",
            "Deleting both intermediate output files BEN_SI19_sub_1_trimmed.fq.gz and BEN_SI19_sub_2_trimmed.fq.gz\n",
            "\n",
            "====================================================================================================\n",
            "\n",
            "Multicore support not enabled. Proceeding with single-core trimming.\n",
            "Path to Cutadapt set as: 'cutadapt' (default)\n",
            "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
            "Cutadapt version: 4.4\n",
            "single-core operation.\n",
            "igzip command line interface 2.31.1\n",
            "igzip detected. Using igzip for decompressing\n",
            "\n",
            "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
            "\n",
            "Output will be written into the directory: /content/trimmed_fastq_files/\n",
            "\n",
            "\n",
            "AUTO-DETECTING ADAPTER TYPE\n",
            "===========================\n",
            "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastq_files/BEN_SI9_sub_1.fq.gz <<)\n",
            "\n",
            "Found perfect matches for the following adapter sequences:\n",
            "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
            "Illumina\t2432\tAGATCGGAAGAGC\t1000000\t0.24\n",
            "Nextera\t15\tCTGTCTCTTATA\t1000000\t0.00\n",
            "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
            "Using Illumina adapter for trimming (count: 2432). Second best hit was Nextera (count: 15)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_SI9_sub_1.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_SI9_sub_1.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_SI9_sub_1_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_SI9_sub_1.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_SI9_sub_1.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 54.468 s (54.468 µs/read; 1.10 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   360,368 (36.0%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   150,000,000 bp\n",
            "Quality-trimmed:                 948,952 bp (0.6%)\n",
            "Total written (filtered):    148,444,303 bp (99.0%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 360368 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 34.2%\n",
            "  C: 26.8%\n",
            "  G: 17.6%\n",
            "  T: 21.3%\n",
            "  none/other: 0.0%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t258970\t250000.0\t0\t258970\n",
            "2\t69927\t62500.0\t0\t69927\n",
            "3\t22556\t15625.0\t0\t22556\n",
            "4\t4576\t3906.2\t0\t4576\n",
            "5\t1038\t976.6\t0\t1038\n",
            "6\t159\t244.1\t0\t159\n",
            "7\t83\t61.0\t0\t83\n",
            "8\t67\t15.3\t0\t67\n",
            "9\t82\t3.8\t0\t65 17\n",
            "10\t103\t1.0\t1\t77 26\n",
            "11\t83\t0.2\t1\t67 16\n",
            "12\t77\t0.1\t1\t69 8\n",
            "13\t58\t0.0\t1\t57 1\n",
            "14\t65\t0.0\t1\t62 3\n",
            "15\t61\t0.0\t1\t59 2\n",
            "16\t63\t0.0\t1\t63\n",
            "17\t70\t0.0\t1\t65 5\n",
            "18\t71\t0.0\t1\t67 4\n",
            "19\t76\t0.0\t1\t73 3\n",
            "20\t65\t0.0\t1\t62 3\n",
            "21\t67\t0.0\t1\t65 2\n",
            "22\t54\t0.0\t1\t51 3\n",
            "23\t55\t0.0\t1\t51 4\n",
            "24\t51\t0.0\t1\t50 1\n",
            "25\t51\t0.0\t1\t46 5\n",
            "26\t55\t0.0\t1\t54 1\n",
            "27\t58\t0.0\t1\t55 3\n",
            "28\t69\t0.0\t1\t65 4\n",
            "29\t42\t0.0\t1\t36 6\n",
            "30\t56\t0.0\t1\t54 2\n",
            "31\t51\t0.0\t1\t48 3\n",
            "32\t47\t0.0\t1\t45 2\n",
            "33\t30\t0.0\t1\t29 1\n",
            "34\t47\t0.0\t1\t45 2\n",
            "35\t36\t0.0\t1\t33 3\n",
            "36\t40\t0.0\t1\t37 3\n",
            "37\t56\t0.0\t1\t52 4\n",
            "38\t28\t0.0\t1\t27 1\n",
            "39\t51\t0.0\t1\t42 9\n",
            "40\t37\t0.0\t1\t37\n",
            "41\t32\t0.0\t1\t30 2\n",
            "42\t33\t0.0\t1\t33\n",
            "43\t50\t0.0\t1\t49 1\n",
            "44\t30\t0.0\t1\t26 4\n",
            "45\t31\t0.0\t1\t29 2\n",
            "46\t25\t0.0\t1\t24 1\n",
            "47\t32\t0.0\t1\t31 1\n",
            "48\t36\t0.0\t1\t36\n",
            "49\t35\t0.0\t1\t34 1\n",
            "50\t35\t0.0\t1\t33 2\n",
            "51\t41\t0.0\t1\t40 1\n",
            "52\t28\t0.0\t1\t25 3\n",
            "53\t24\t0.0\t1\t23 1\n",
            "54\t24\t0.0\t1\t23 1\n",
            "55\t29\t0.0\t1\t28 1\n",
            "56\t22\t0.0\t1\t20 2\n",
            "57\t24\t0.0\t1\t19 5\n",
            "58\t19\t0.0\t1\t18 1\n",
            "59\t25\t0.0\t1\t22 3\n",
            "60\t27\t0.0\t1\t26 1\n",
            "61\t15\t0.0\t1\t14 1\n",
            "62\t14\t0.0\t1\t12 2\n",
            "63\t19\t0.0\t1\t18 1\n",
            "64\t17\t0.0\t1\t17\n",
            "65\t21\t0.0\t1\t19 2\n",
            "66\t13\t0.0\t1\t11 2\n",
            "67\t8\t0.0\t1\t7 1\n",
            "68\t11\t0.0\t1\t10 1\n",
            "69\t17\t0.0\t1\t15 2\n",
            "70\t18\t0.0\t1\t16 2\n",
            "71\t26\t0.0\t1\t26\n",
            "72\t15\t0.0\t1\t14 1\n",
            "73\t20\t0.0\t1\t17 3\n",
            "74\t15\t0.0\t1\t13 2\n",
            "75\t19\t0.0\t1\t13 6\n",
            "76\t72\t0.0\t1\t66 6\n",
            "77\t33\t0.0\t1\t28 5\n",
            "78\t15\t0.0\t1\t14 1\n",
            "79\t10\t0.0\t1\t7 3\n",
            "80\t5\t0.0\t1\t5\n",
            "81\t4\t0.0\t1\t4\n",
            "82\t2\t0.0\t1\t2\n",
            "83\t9\t0.0\t1\t5 4\n",
            "84\t4\t0.0\t1\t2 2\n",
            "85\t7\t0.0\t1\t6 1\n",
            "86\t4\t0.0\t1\t2 2\n",
            "87\t3\t0.0\t1\t2 1\n",
            "88\t3\t0.0\t1\t2 1\n",
            "89\t3\t0.0\t1\t3\n",
            "90\t7\t0.0\t1\t4 3\n",
            "91\t3\t0.0\t1\t3\n",
            "92\t1\t0.0\t1\t1\n",
            "94\t3\t0.0\t1\t3\n",
            "95\t6\t0.0\t1\t2 4\n",
            "96\t1\t0.0\t1\t0 1\n",
            "97\t3\t0.0\t1\t2 1\n",
            "98\t4\t0.0\t1\t2 2\n",
            "99\t6\t0.0\t1\t3 3\n",
            "100\t7\t0.0\t1\t3 4\n",
            "101\t1\t0.0\t1\t0 1\n",
            "102\t2\t0.0\t1\t1 1\n",
            "103\t3\t0.0\t1\t1 2\n",
            "104\t3\t0.0\t1\t1 2\n",
            "105\t3\t0.0\t1\t1 2\n",
            "106\t3\t0.0\t1\t3\n",
            "107\t2\t0.0\t1\t1 1\n",
            "108\t2\t0.0\t1\t0 2\n",
            "109\t2\t0.0\t1\t1 1\n",
            "110\t1\t0.0\t1\t1\n",
            "111\t2\t0.0\t1\t0 2\n",
            "112\t1\t0.0\t1\t1\n",
            "113\t3\t0.0\t1\t1 2\n",
            "114\t1\t0.0\t1\t0 1\n",
            "115\t2\t0.0\t1\t1 1\n",
            "116\t2\t0.0\t1\t2\n",
            "118\t1\t0.0\t1\t1\n",
            "119\t4\t0.0\t1\t2 2\n",
            "120\t2\t0.0\t1\t2\n",
            "123\t4\t0.0\t1\t2 2\n",
            "124\t3\t0.0\t1\t2 1\n",
            "125\t1\t0.0\t1\t0 1\n",
            "127\t1\t0.0\t1\t1\n",
            "128\t3\t0.0\t1\t3\n",
            "129\t3\t0.0\t1\t1 2\n",
            "130\t1\t0.0\t1\t1\n",
            "132\t1\t0.0\t1\t1\n",
            "133\t1\t0.0\t1\t1\n",
            "134\t1\t0.0\t1\t0 1\n",
            "135\t1\t0.0\t1\t0 1\n",
            "139\t1\t0.0\t1\t0 1\n",
            "140\t1\t0.0\t1\t0 1\n",
            "141\t4\t0.0\t1\t2 2\n",
            "144\t1\t0.0\t1\t0 1\n",
            "145\t1\t0.0\t1\t0 1\n",
            "148\t1\t0.0\t1\t0 1\n",
            "149\t1\t0.0\t1\t0 1\n",
            "150\t2\t0.0\t1\t0 2\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_SI9_sub_1.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/BEN_SI9_sub_2.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/BEN_SI9_sub_2.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 1\n",
            "Writing final adapter and quality trimmed output to BEN_SI9_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/BEN_SI9_sub_2.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/BEN_SI9_sub_2.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 60.020 s (60.020 µs/read; 1.00 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   351,796 (35.2%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   150,000,000 bp\n",
            "Quality-trimmed:               2,318,066 bp (1.5%)\n",
            "Total written (filtered):    147,059,976 bp (98.0%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 351796 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 34.2%\n",
            "  C: 26.8%\n",
            "  G: 17.3%\n",
            "  T: 21.7%\n",
            "  none/other: 0.0%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t251933\t250000.0\t0\t251933\n",
            "2\t69366\t62500.0\t0\t69366\n",
            "3\t21454\t15625.0\t0\t21454\n",
            "4\t4363\t3906.2\t0\t4363\n",
            "5\t973\t976.6\t0\t973\n",
            "6\t151\t244.1\t0\t151\n",
            "7\t94\t61.0\t0\t94\n",
            "8\t76\t15.3\t0\t76\n",
            "9\t84\t3.8\t0\t61 23\n",
            "10\t101\t1.0\t1\t80 21\n",
            "11\t85\t0.2\t1\t68 17\n",
            "12\t81\t0.1\t1\t72 9\n",
            "13\t64\t0.0\t1\t58 6\n",
            "14\t95\t0.0\t1\t81 14\n",
            "15\t52\t0.0\t1\t48 4\n",
            "16\t72\t0.0\t1\t70 2\n",
            "17\t81\t0.0\t1\t74 7\n",
            "18\t58\t0.0\t1\t56 2\n",
            "19\t85\t0.0\t1\t81 4\n",
            "20\t60\t0.0\t1\t60\n",
            "21\t62\t0.0\t1\t62\n",
            "22\t53\t0.0\t1\t50 3\n",
            "23\t61\t0.0\t1\t57 4\n",
            "24\t71\t0.0\t1\t65 6\n",
            "25\t41\t0.0\t1\t35 6\n",
            "26\t53\t0.0\t1\t44 9\n",
            "27\t60\t0.0\t1\t54 6\n",
            "28\t68\t0.0\t1\t57 11\n",
            "29\t44\t0.0\t1\t37 7\n",
            "30\t64\t0.0\t1\t54 10\n",
            "31\t48\t0.0\t1\t45 3\n",
            "32\t51\t0.0\t1\t40 11\n",
            "33\t34\t0.0\t1\t30 4\n",
            "34\t42\t0.0\t1\t35 7\n",
            "35\t37\t0.0\t1\t27 10\n",
            "36\t49\t0.0\t1\t39 10\n",
            "37\t48\t0.0\t1\t37 11\n",
            "38\t31\t0.0\t1\t26 5\n",
            "39\t55\t0.0\t1\t50 5\n",
            "40\t34\t0.0\t1\t33 1\n",
            "41\t34\t0.0\t1\t34\n",
            "42\t42\t0.0\t1\t42\n",
            "43\t36\t0.0\t1\t34 2\n",
            "44\t38\t0.0\t1\t36 2\n",
            "45\t39\t0.0\t1\t37 2\n",
            "46\t30\t0.0\t1\t28 2\n",
            "47\t41\t0.0\t1\t36 5\n",
            "48\t34\t0.0\t1\t32 2\n",
            "49\t34\t0.0\t1\t32 2\n",
            "50\t38\t0.0\t1\t36 2\n",
            "51\t42\t0.0\t1\t40 2\n",
            "52\t29\t0.0\t1\t28 1\n",
            "53\t26\t0.0\t1\t24 2\n",
            "54\t35\t0.0\t1\t34 1\n",
            "55\t22\t0.0\t1\t20 2\n",
            "56\t26\t0.0\t1\t24 2\n",
            "57\t30\t0.0\t1\t25 5\n",
            "58\t22\t0.0\t1\t20 2\n",
            "59\t26\t0.0\t1\t24 2\n",
            "60\t37\t0.0\t1\t33 4\n",
            "61\t17\t0.0\t1\t15 2\n",
            "62\t15\t0.0\t1\t14 1\n",
            "63\t54\t0.0\t1\t49 5\n",
            "64\t11\t0.0\t1\t11\n",
            "65\t21\t0.0\t1\t20 1\n",
            "66\t6\t0.0\t1\t5 1\n",
            "67\t15\t0.0\t1\t13 2\n",
            "68\t13\t0.0\t1\t11 2\n",
            "69\t17\t0.0\t1\t14 3\n",
            "70\t25\t0.0\t1\t23 2\n",
            "71\t24\t0.0\t1\t23 1\n",
            "72\t22\t0.0\t1\t20 2\n",
            "73\t21\t0.0\t1\t18 3\n",
            "74\t12\t0.0\t1\t10 2\n",
            "75\t16\t0.0\t1\t11 5\n",
            "76\t17\t0.0\t1\t15 2\n",
            "77\t19\t0.0\t1\t17 2\n",
            "78\t22\t0.0\t1\t19 3\n",
            "79\t37\t0.0\t1\t24 13\n",
            "80\t135\t0.0\t1\t120 15\n",
            "81\t36\t0.0\t1\t34 2\n",
            "82\t15\t0.0\t1\t14 1\n",
            "83\t4\t0.0\t1\t4\n",
            "84\t2\t0.0\t1\t0 2\n",
            "85\t9\t0.0\t1\t7 2\n",
            "86\t4\t0.0\t1\t1 3\n",
            "87\t5\t0.0\t1\t4 1\n",
            "88\t7\t0.0\t1\t6 1\n",
            "89\t3\t0.0\t1\t1 2\n",
            "90\t6\t0.0\t1\t5 1\n",
            "91\t3\t0.0\t1\t3\n",
            "92\t6\t0.0\t1\t3 3\n",
            "93\t7\t0.0\t1\t5 2\n",
            "94\t4\t0.0\t1\t1 3\n",
            "95\t2\t0.0\t1\t1 1\n",
            "96\t6\t0.0\t1\t6\n",
            "97\t5\t0.0\t1\t2 3\n",
            "98\t3\t0.0\t1\t1 2\n",
            "99\t2\t0.0\t1\t2\n",
            "100\t10\t0.0\t1\t5 5\n",
            "101\t5\t0.0\t1\t0 5\n",
            "102\t6\t0.0\t1\t4 2\n",
            "103\t3\t0.0\t1\t2 1\n",
            "104\t4\t0.0\t1\t3 1\n",
            "105\t4\t0.0\t1\t1 3\n",
            "106\t7\t0.0\t1\t6 1\n",
            "107\t7\t0.0\t1\t4 3\n",
            "108\t7\t0.0\t1\t4 3\n",
            "109\t5\t0.0\t1\t2 3\n",
            "110\t9\t0.0\t1\t5 4\n",
            "111\t8\t0.0\t1\t7 1\n",
            "112\t2\t0.0\t1\t1 1\n",
            "113\t3\t0.0\t1\t0 3\n",
            "114\t4\t0.0\t1\t2 2\n",
            "115\t2\t0.0\t1\t2\n",
            "116\t1\t0.0\t1\t0 1\n",
            "117\t6\t0.0\t1\t4 2\n",
            "118\t4\t0.0\t1\t3 1\n",
            "119\t2\t0.0\t1\t1 1\n",
            "120\t4\t0.0\t1\t2 2\n",
            "121\t3\t0.0\t1\t2 1\n",
            "122\t3\t0.0\t1\t3\n",
            "123\t3\t0.0\t1\t3\n",
            "124\t6\t0.0\t1\t2 4\n",
            "125\t3\t0.0\t1\t0 3\n",
            "126\t3\t0.0\t1\t2 1\n",
            "128\t6\t0.0\t1\t3 3\n",
            "129\t9\t0.0\t1\t2 7\n",
            "131\t2\t0.0\t1\t1 1\n",
            "132\t2\t0.0\t1\t1 1\n",
            "134\t1\t0.0\t1\t1\n",
            "135\t3\t0.0\t1\t1 2\n",
            "137\t1\t0.0\t1\t0 1\n",
            "140\t1\t0.0\t1\t0 1\n",
            "141\t1\t0.0\t1\t0 1\n",
            "146\t3\t0.0\t1\t0 3\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/BEN_SI9_sub_2.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Validate paired-end files BEN_SI9_sub_1_trimmed.fq.gz and BEN_SI9_sub_2_trimmed.fq.gz\n",
            "file_1: BEN_SI9_sub_1_trimmed.fq.gz, file_2: BEN_SI9_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            ">>>>> Now validing the length of the 2 paired-end infiles: BEN_SI9_sub_1_trimmed.fq.gz and BEN_SI9_sub_2_trimmed.fq.gz <<<<<\n",
            "Writing validated paired-end Read 1 reads to BEN_SI9_sub_1_val_1.fq.gz\n",
            "Writing validated paired-end Read 2 reads to BEN_SI9_sub_2_val_2.fq.gz\n",
            "\n",
            "Total number of sequences analysed: 1000000\n",
            "\n",
            "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1957 (0.20%)\n",
            "\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_SI9_sub_1_val_1.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 5% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 10% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 15% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 20% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 25% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 30% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 35% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 40% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 45% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 50% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 55% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 60% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 65% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 70% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 75% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 80% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 85% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 90% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Approx 95% complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "Analysis complete for BEN_SI9_sub_1_val_1.fq.gz\n",
            "\n",
            "  >>> Now running FastQC on the validated data BEN_SI9_sub_2_val_2.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 5% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 10% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 15% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 20% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 25% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 30% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 35% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 40% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 45% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 50% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 55% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 60% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 65% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 70% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 75% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 80% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 85% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 90% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Approx 95% complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Analysis complete for BEN_SI9_sub_2_val_2.fq.gz\n",
            "Deleting both intermediate output files BEN_SI9_sub_1_trimmed.fq.gz and BEN_SI9_sub_2_trimmed.fq.gz\n",
            "\n",
            "====================================================================================================\n",
            "\n",
            "Multicore support not enabled. Proceeding with single-core trimming.\n",
            "Path to Cutadapt set as: 'cutadapt' (default)\n",
            "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
            "Cutadapt version: 4.4\n",
            "single-core operation.\n",
            "igzip command line interface 2.31.1\n",
            "igzip detected. Using igzip for decompressing\n",
            "\n",
            "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
            "\n",
            "Output will be written into the directory: /content/trimmed_fastq_files/\n",
            "\n",
            "\n",
            "AUTO-DETECTING ADAPTER TYPE\n",
            "===========================\n",
            "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> fastq_files/LGS1_sub_1.fq.gz <<)\n",
            "\n",
            "Found perfect matches for the following adapter sequences:\n",
            "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
            "Illumina\t7872\tAGATCGGAAGAGC\t1000000\t0.79\n",
            "Nextera\t12\tCTGTCTCTTATA\t1000000\t0.00\n",
            "smallRNA\t1\tTGGAATTCTCGG\t1000000\t0.00\n",
            "Using Illumina adapter for trimming (count: 7872). Second best hit was Nextera (count: 12)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/LGS1_sub_1.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/LGS1_sub_1.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j 1\n",
            "Writing final adapter and quality trimmed output to LGS1_sub_1_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/LGS1_sub_1.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/LGS1_sub_1.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 55.255 s (55.255 µs/read; 1.09 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   366,553 (36.7%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   150,000,000 bp\n",
            "Quality-trimmed:                 580,935 bp (0.4%)\n",
            "Total written (filtered):    148,622,332 bp (99.1%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 366553 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 32.9%\n",
            "  C: 27.7%\n",
            "  G: 18.8%\n",
            "  T: 20.6%\n",
            "  none/other: 0.0%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t252024\t250000.0\t0\t252024\n",
            "2\t72127\t62500.0\t0\t72127\n",
            "3\t23370\t15625.0\t0\t23370\n",
            "4\t5016\t3906.2\t0\t5016\n",
            "5\t1691\t976.6\t0\t1691\n",
            "6\t727\t244.1\t0\t727\n",
            "7\t600\t61.0\t0\t600\n",
            "8\t559\t15.3\t0\t559\n",
            "9\t571\t3.8\t0\t546 25\n",
            "10\t539\t1.0\t1\t494 45\n",
            "11\t503\t0.2\t1\t480 23\n",
            "12\t470\t0.1\t1\t454 16\n",
            "13\t406\t0.0\t1\t390 16\n",
            "14\t391\t0.0\t1\t377 14\n",
            "15\t376\t0.0\t1\t357 19\n",
            "16\t374\t0.0\t1\t361 13\n",
            "17\t403\t0.0\t1\t382 21\n",
            "18\t357\t0.0\t1\t342 15\n",
            "19\t320\t0.0\t1\t306 14\n",
            "20\t324\t0.0\t1\t313 11\n",
            "21\t280\t0.0\t1\t268 12\n",
            "22\t250\t0.0\t1\t239 11\n",
            "23\t231\t0.0\t1\t222 9\n",
            "24\t228\t0.0\t1\t218 10\n",
            "25\t206\t0.0\t1\t203 3\n",
            "26\t212\t0.0\t1\t203 9\n",
            "27\t225\t0.0\t1\t209 16\n",
            "28\t201\t0.0\t1\t194 7\n",
            "29\t187\t0.0\t1\t181 6\n",
            "30\t168\t0.0\t1\t160 8\n",
            "31\t150\t0.0\t1\t142 8\n",
            "32\t151\t0.0\t1\t143 8\n",
            "33\t131\t0.0\t1\t127 4\n",
            "34\t102\t0.0\t1\t96 6\n",
            "35\t111\t0.0\t1\t105 6\n",
            "36\t133\t0.0\t1\t132 1\n",
            "37\t120\t0.0\t1\t115 5\n",
            "38\t117\t0.0\t1\t116 1\n",
            "39\t130\t0.0\t1\t128 2\n",
            "40\t121\t0.0\t1\t118 3\n",
            "41\t113\t0.0\t1\t110 3\n",
            "42\t93\t0.0\t1\t89 4\n",
            "43\t83\t0.0\t1\t80 3\n",
            "44\t68\t0.0\t1\t65 3\n",
            "45\t88\t0.0\t1\t85 3\n",
            "46\t65\t0.0\t1\t61 4\n",
            "47\t74\t0.0\t1\t71 3\n",
            "48\t61\t0.0\t1\t60 1\n",
            "49\t51\t0.0\t1\t48 3\n",
            "50\t55\t0.0\t1\t54 1\n",
            "51\t40\t0.0\t1\t39 1\n",
            "52\t56\t0.0\t1\t55 1\n",
            "53\t34\t0.0\t1\t34\n",
            "54\t40\t0.0\t1\t38 2\n",
            "55\t48\t0.0\t1\t45 3\n",
            "56\t40\t0.0\t1\t39 1\n",
            "57\t30\t0.0\t1\t28 2\n",
            "58\t35\t0.0\t1\t34 1\n",
            "59\t45\t0.0\t1\t44 1\n",
            "60\t38\t0.0\t1\t37 1\n",
            "61\t29\t0.0\t1\t28 1\n",
            "62\t28\t0.0\t1\t28\n",
            "63\t24\t0.0\t1\t24\n",
            "64\t27\t0.0\t1\t25 2\n",
            "65\t37\t0.0\t1\t31 6\n",
            "66\t28\t0.0\t1\t26 2\n",
            "67\t23\t0.0\t1\t22 1\n",
            "68\t23\t0.0\t1\t23\n",
            "69\t27\t0.0\t1\t25 2\n",
            "70\t34\t0.0\t1\t28 6\n",
            "71\t87\t0.0\t1\t29 58\n",
            "72\t160\t0.0\t1\t134 26\n",
            "73\t36\t0.0\t1\t32 4\n",
            "74\t21\t0.0\t1\t12 9\n",
            "75\t3\t0.0\t1\t2 1\n",
            "76\t7\t0.0\t1\t4 3\n",
            "77\t5\t0.0\t1\t4 1\n",
            "78\t7\t0.0\t1\t5 2\n",
            "79\t9\t0.0\t1\t7 2\n",
            "80\t6\t0.0\t1\t4 2\n",
            "81\t5\t0.0\t1\t2 3\n",
            "82\t7\t0.0\t1\t4 3\n",
            "83\t8\t0.0\t1\t6 2\n",
            "84\t4\t0.0\t1\t4\n",
            "85\t3\t0.0\t1\t1 2\n",
            "86\t1\t0.0\t1\t1\n",
            "87\t2\t0.0\t1\t2\n",
            "88\t1\t0.0\t1\t1\n",
            "89\t3\t0.0\t1\t2 1\n",
            "90\t3\t0.0\t1\t3\n",
            "91\t4\t0.0\t1\t3 1\n",
            "92\t3\t0.0\t1\t2 1\n",
            "93\t3\t0.0\t1\t2 1\n",
            "94\t1\t0.0\t1\t1\n",
            "95\t3\t0.0\t1\t3\n",
            "96\t4\t0.0\t1\t2 2\n",
            "97\t3\t0.0\t1\t1 2\n",
            "98\t4\t0.0\t1\t1 3\n",
            "99\t2\t0.0\t1\t1 1\n",
            "100\t1\t0.0\t1\t0 1\n",
            "101\t3\t0.0\t1\t1 2\n",
            "102\t4\t0.0\t1\t3 1\n",
            "103\t3\t0.0\t1\t2 1\n",
            "104\t2\t0.0\t1\t1 1\n",
            "105\t1\t0.0\t1\t1\n",
            "106\t1\t0.0\t1\t0 1\n",
            "108\t4\t0.0\t1\t3 1\n",
            "109\t2\t0.0\t1\t1 1\n",
            "110\t2\t0.0\t1\t1 1\n",
            "111\t4\t0.0\t1\t3 1\n",
            "112\t2\t0.0\t1\t2\n",
            "113\t2\t0.0\t1\t0 2\n",
            "114\t3\t0.0\t1\t2 1\n",
            "115\t4\t0.0\t1\t2 2\n",
            "116\t2\t0.0\t1\t0 2\n",
            "117\t1\t0.0\t1\t1\n",
            "118\t1\t0.0\t1\t0 1\n",
            "119\t2\t0.0\t1\t0 2\n",
            "121\t2\t0.0\t1\t1 1\n",
            "122\t1\t0.0\t1\t0 1\n",
            "123\t1\t0.0\t1\t0 1\n",
            "125\t1\t0.0\t1\t0 1\n",
            "127\t1\t0.0\t1\t0 1\n",
            "128\t1\t0.0\t1\t1\n",
            "129\t1\t0.0\t1\t0 1\n",
            "130\t1\t0.0\t1\t1\n",
            "132\t1\t0.0\t1\t0 1\n",
            "133\t1\t0.0\t1\t0 1\n",
            "135\t1\t0.0\t1\t0 1\n",
            "136\t1\t0.0\t1\t0 1\n",
            "138\t2\t0.0\t1\t0 2\n",
            "141\t1\t0.0\t1\t0 1\n",
            "143\t2\t0.0\t1\t0 2\n",
            "144\t1\t0.0\t1\t0 1\n",
            "145\t3\t0.0\t1\t0 3\n",
            "146\t3\t0.0\t1\t0 3\n",
            "147\t1\t0.0\t1\t0 1\n",
            "148\t2\t0.0\t1\t0 2\n",
            "149\t3\t0.0\t1\t0 3\n",
            "150\t34\t0.0\t1\t0 34\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/LGS1_sub_1.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Writing report to '/content/trimmed_fastq_files/LGS1_sub_2.fq.gz_trimming_report.txt'\n",
            "\n",
            "SUMMARISING RUN PARAMETERS\n",
            "==========================\n",
            "Input filename: fastq_files/LGS1_sub_2.fq.gz\n",
            "Trimming mode: paired-end\n",
            "Trim Galore version: 0.6.10\n",
            "Cutadapt version: 4.4\n",
            "Number of cores used for trimming: 1\n",
            "Quality Phred score cutoff: 20\n",
            "Quality encoding type selected: ASCII+33\n",
            "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)\n",
            "Maximum trimming error rate: 0.1 (default)\n",
            "Minimum required adapter overlap (stringency): 1 bp\n",
            "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
            "Running FastQC on the data once trimming has completed\n",
            "Output file(s) will be GZIP compressed\n",
            "\n",
            "Cutadapt seems to be fairly up-to-date (version 4.4). Setting -j -j 1\n",
            "Writing final adapter and quality trimmed output to LGS1_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            "  >>> Now performing quality (cutoff '-q 20') and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file fastq_files/LGS1_sub_2.fq.gz <<< \n",
            "This is cutadapt 4.4 with Python 3.7.12\n",
            "Command line parameters: -j 1 -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC fastq_files/LGS1_sub_2.fq.gz\n",
            "Processing single-end reads on 1 core ...\n",
            "Finished in 57.381 s (57.381 µs/read; 1.05 M reads/minute).\n",
            "\n",
            "=== Summary ===\n",
            "\n",
            "Total reads processed:               1,000,000\n",
            "Reads with adapters:                   364,759 (36.5%)\n",
            "Reads written (passing filters):     1,000,000 (100.0%)\n",
            "\n",
            "Total basepairs processed:   150,000,000 bp\n",
            "Quality-trimmed:               1,619,060 bp (1.1%)\n",
            "Total written (filtered):    147,583,589 bp (98.4%)\n",
            "\n",
            "=== Adapter 1 ===\n",
            "\n",
            "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 364759 times\n",
            "\n",
            "Minimum overlap: 1\n",
            "No. of allowed errors:\n",
            "1-9 bp: 0; 10-13 bp: 1\n",
            "\n",
            "Bases preceding removed adapters:\n",
            "  A: 32.9%\n",
            "  C: 27.8%\n",
            "  G: 18.5%\n",
            "  T: 20.7%\n",
            "  none/other: 0.1%\n",
            "\n",
            "Overview of removed sequences\n",
            "length\tcount\texpect\tmax.err\terror counts\n",
            "1\t250249\t250000.0\t0\t250249\n",
            "2\t72346\t62500.0\t0\t72346\n",
            "3\t22936\t15625.0\t0\t22936\n",
            "4\t5170\t3906.2\t0\t5170\n",
            "5\t1574\t976.6\t0\t1574\n",
            "6\t724\t244.1\t0\t724\n",
            "7\t601\t61.0\t0\t601\n",
            "8\t566\t15.3\t0\t566\n",
            "9\t585\t3.8\t0\t568 17\n",
            "10\t539\t1.0\t1\t502 37\n",
            "11\t518\t0.2\t1\t492 26\n",
            "12\t485\t0.1\t1\t461 24\n",
            "13\t402\t0.0\t1\t382 20\n",
            "14\t459\t0.0\t1\t446 13\n",
            "15\t356\t0.0\t1\t346 10\n",
            "16\t382\t0.0\t1\t366 16\n",
            "17\t373\t0.0\t1\t357 16\n",
            "18\t302\t0.0\t1\t287 15\n",
            "19\t365\t0.0\t1\t355 10\n",
            "20\t305\t0.0\t1\t298 7\n",
            "21\t278\t0.0\t1\t270 8\n",
            "22\t252\t0.0\t1\t246 6\n",
            "23\t238\t0.0\t1\t230 8\n",
            "24\t239\t0.0\t1\t232 7\n",
            "25\t196\t0.0\t1\t188 8\n",
            "26\t210\t0.0\t1\t201 9\n",
            "27\t217\t0.0\t1\t210 7\n",
            "28\t212\t0.0\t1\t205 7\n",
            "29\t185\t0.0\t1\t175 10\n",
            "30\t189\t0.0\t1\t181 8\n",
            "31\t130\t0.0\t1\t127 3\n",
            "32\t159\t0.0\t1\t152 7\n",
            "33\t128\t0.0\t1\t123 5\n",
            "34\t112\t0.0\t1\t107 5\n",
            "35\t116\t0.0\t1\t110 6\n",
            "36\t127\t0.0\t1\t124 3\n",
            "37\t122\t0.0\t1\t116 6\n",
            "38\t115\t0.0\t1\t109 6\n",
            "39\t129\t0.0\t1\t124 5\n",
            "40\t121\t0.0\t1\t119 2\n",
            "41\t119\t0.0\t1\t116 3\n",
            "42\t87\t0.0\t1\t83 4\n",
            "43\t80\t0.0\t1\t76 4\n",
            "44\t76\t0.0\t1\t75 1\n",
            "45\t81\t0.0\t1\t81\n",
            "46\t73\t0.0\t1\t70 3\n",
            "47\t76\t0.0\t1\t75 1\n",
            "48\t69\t0.0\t1\t64 5\n",
            "49\t52\t0.0\t1\t48 4\n",
            "50\t45\t0.0\t1\t45\n",
            "51\t60\t0.0\t1\t55 5\n",
            "52\t50\t0.0\t1\t48 2\n",
            "53\t36\t0.0\t1\t36\n",
            "54\t33\t0.0\t1\t32 1\n",
            "55\t51\t0.0\t1\t48 3\n",
            "56\t37\t0.0\t1\t36 1\n",
            "57\t44\t0.0\t1\t38 6\n",
            "58\t34\t0.0\t1\t33 1\n",
            "59\t39\t0.0\t1\t37 2\n",
            "60\t46\t0.0\t1\t43 3\n",
            "61\t31\t0.0\t1\t31\n",
            "62\t32\t0.0\t1\t30 2\n",
            "63\t96\t0.0\t1\t32 64\n",
            "64\t317\t0.0\t1\t285 32\n",
            "65\t84\t0.0\t1\t68 16\n",
            "66\t36\t0.0\t1\t28 8\n",
            "67\t15\t0.0\t1\t8 7\n",
            "68\t10\t0.0\t1\t10\n",
            "69\t8\t0.0\t1\t5 3\n",
            "70\t8\t0.0\t1\t6 2\n",
            "71\t7\t0.0\t1\t7\n",
            "72\t7\t0.0\t1\t5 2\n",
            "73\t5\t0.0\t1\t4 1\n",
            "74\t9\t0.0\t1\t7 2\n",
            "75\t4\t0.0\t1\t4\n",
            "76\t6\t0.0\t1\t4 2\n",
            "77\t5\t0.0\t1\t5\n",
            "78\t2\t0.0\t1\t2\n",
            "79\t4\t0.0\t1\t3 1\n",
            "80\t4\t0.0\t1\t3 1\n",
            "81\t7\t0.0\t1\t3 4\n",
            "82\t3\t0.0\t1\t3\n",
            "83\t6\t0.0\t1\t6\n",
            "84\t5\t0.0\t1\t3 2\n",
            "85\t5\t0.0\t1\t3 2\n",
            "86\t1\t0.0\t1\t0 1\n",
            "87\t3\t0.0\t1\t1 2\n",
            "88\t4\t0.0\t1\t3 1\n",
            "89\t5\t0.0\t1\t3 2\n",
            "90\t3\t0.0\t1\t2 1\n",
            "91\t4\t0.0\t1\t2 2\n",
            "92\t1\t0.0\t1\t0 1\n",
            "93\t4\t0.0\t1\t2 2\n",
            "94\t2\t0.0\t1\t1 1\n",
            "95\t4\t0.0\t1\t4\n",
            "96\t2\t0.0\t1\t2\n",
            "97\t1\t0.0\t1\t1\n",
            "98\t8\t0.0\t1\t7 1\n",
            "99\t3\t0.0\t1\t2 1\n",
            "101\t2\t0.0\t1\t1 1\n",
            "102\t3\t0.0\t1\t1 2\n",
            "103\t2\t0.0\t1\t1 1\n",
            "104\t3\t0.0\t1\t1 2\n",
            "105\t3\t0.0\t1\t1 2\n",
            "106\t2\t0.0\t1\t1 1\n",
            "107\t3\t0.0\t1\t2 1\n",
            "108\t3\t0.0\t1\t1 2\n",
            "109\t4\t0.0\t1\t3 1\n",
            "110\t2\t0.0\t1\t1 1\n",
            "111\t2\t0.0\t1\t2\n",
            "112\t3\t0.0\t1\t1 2\n",
            "113\t2\t0.0\t1\t2\n",
            "114\t1\t0.0\t1\t1\n",
            "115\t6\t0.0\t1\t3 3\n",
            "117\t1\t0.0\t1\t1\n",
            "120\t1\t0.0\t1\t1\n",
            "122\t3\t0.0\t1\t0 3\n",
            "123\t1\t0.0\t1\t1\n",
            "124\t1\t0.0\t1\t0 1\n",
            "125\t1\t0.0\t1\t1\n",
            "126\t1\t0.0\t1\t0 1\n",
            "129\t1\t0.0\t1\t0 1\n",
            "130\t1\t0.0\t1\t1\n",
            "134\t1\t0.0\t1\t0 1\n",
            "137\t1\t0.0\t1\t0 1\n",
            "143\t1\t0.0\t1\t0 1\n",
            "146\t2\t0.0\t1\t0 2\n",
            "147\t1\t0.0\t1\t0 1\n",
            "148\t1\t0.0\t1\t0 1\n",
            "149\t3\t0.0\t1\t0 3\n",
            "150\t36\t0.0\t1\t0 36\n",
            "\n",
            "RUN STATISTICS FOR INPUT FILE: fastq_files/LGS1_sub_2.fq.gz\n",
            "=============================================\n",
            "1000000 sequences processed in total\n",
            "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
            "\n",
            "Validate paired-end files LGS1_sub_1_trimmed.fq.gz and LGS1_sub_2_trimmed.fq.gz\n",
            "file_1: LGS1_sub_1_trimmed.fq.gz, file_2: LGS1_sub_2_trimmed.fq.gz\n",
            "\n",
            "\n",
            ">>>>> Now validing the length of the 2 paired-end infiles: LGS1_sub_1_trimmed.fq.gz and LGS1_sub_2_trimmed.fq.gz <<<<<\n",
            "Writing validated paired-end Read 1 reads to LGS1_sub_1_val_1.fq.gz\n",
            "Writing validated paired-end Read 2 reads to LGS1_sub_2_val_2.fq.gz\n",
            "\n",
            "Total number of sequences analysed: 1000000\n",
            "\n",
            "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 1660 (0.17%)\n",
            "\n",
            "\n",
            "  >>> Now running FastQC on the validated data LGS1_sub_1_val_1.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of LGS1_sub_1_val_1.fq.gz\n",
            "Approx 5% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 10% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 15% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 20% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 25% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 30% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 35% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 40% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 45% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 50% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 55% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 60% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 65% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 70% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 75% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 80% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 85% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 90% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Approx 95% complete for LGS1_sub_1_val_1.fq.gz\n",
            "Analysis complete for LGS1_sub_1_val_1.fq.gz\n",
            "\n",
            "  >>> Now running FastQC on the validated data LGS1_sub_2_val_2.fq.gz<<<\n",
            "\n",
            "application/gzip\n",
            "Started analysis of LGS1_sub_2_val_2.fq.gz\n",
            "Approx 5% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 10% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 15% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 20% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 25% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 30% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 35% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 40% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 45% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 50% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 55% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 60% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 65% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 70% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 75% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 80% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 85% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 90% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Approx 95% complete for LGS1_sub_2_val_2.fq.gz\n",
            "Analysis complete for LGS1_sub_2_val_2.fq.gz\n",
            "Deleting both intermediate output files LGS1_sub_1_trimmed.fq.gz and LGS1_sub_2_trimmed.fq.gz\n",
            "\n",
            "====================================================================================================\n",
            "\n",
            "BEN_NW13_sub_1.fq.gz_trimming_report.txt\n",
            "BEN_NW13_sub_1_val_1_fastqc.html\n",
            "BEN_NW13_sub_1_val_1_fastqc.zip\n",
            "BEN_NW13_sub_1_val_1.fq.gz\n",
            "BEN_NW13_sub_2.fq.gz_trimming_report.txt\n",
            "BEN_NW13_sub_2_val_2_fastqc.html\n",
            "BEN_NW13_sub_2_val_2_fastqc.zip\n",
            "BEN_NW13_sub_2_val_2.fq.gz\n",
            "BEN_SI18_sub_1.fq.gz_trimming_report.txt\n",
            "BEN_SI18_sub_1_val_1_fastqc.html\n",
            "BEN_SI18_sub_1_val_1_fastqc.zip\n",
            "BEN_SI18_sub_1_val_1.fq.gz\n",
            "BEN_SI18_sub_2.fq.gz_trimming_report.txt\n",
            "BEN_SI18_sub_2_val_2_fastqc.html\n",
            "BEN_SI18_sub_2_val_2_fastqc.zip\n",
            "BEN_SI18_sub_2_val_2.fq.gz\n",
            "BEN_SI19_sub_1.fq.gz_trimming_report.txt\n",
            "BEN_SI19_sub_1_val_1_fastqc.html\n",
            "BEN_SI19_sub_1_val_1_fastqc.zip\n",
            "BEN_SI19_sub_1_val_1.fq.gz\n",
            "BEN_SI19_sub_2.fq.gz_trimming_report.txt\n",
            "BEN_SI19_sub_2_val_2_fastqc.html\n",
            "BEN_SI19_sub_2_val_2_fastqc.zip\n",
            "BEN_SI19_sub_2_val_2.fq.gz\n",
            "BEN_SI9_sub_1.fq.gz_trimming_report.txt\n",
            "BEN_SI9_sub_1_val_1_fastqc.html\n",
            "BEN_SI9_sub_1_val_1_fastqc.zip\n",
            "BEN_SI9_sub_1_val_1.fq.gz\n",
            "BEN_SI9_sub_2.fq.gz_trimming_report.txt\n",
            "BEN_SI9_sub_2_val_2_fastqc.html\n",
            "BEN_SI9_sub_2_val_2_fastqc.zip\n",
            "BEN_SI9_sub_2_val_2.fq.gz\n",
            "LGS1_sub_1.fq.gz_trimming_report.txt\n",
            "LGS1_sub_1_val_1_fastqc.html\n",
            "LGS1_sub_1_val_1_fastqc.zip\n",
            "LGS1_sub_1_val_1.fq.gz\n",
            "LGS1_sub_2.fq.gz_trimming_report.txt\n",
            "LGS1_sub_2_val_2_fastqc.html\n",
            "LGS1_sub_2_val_2_fastqc.zip\n",
            "LGS1_sub_2_val_2.fq.gz\n",
            "CPU times: user 1.87 s, sys: 270 ms, total: 2.14 s\n",
            "Wall time: 18min 45s\n"
          ]
        }
      ]
    }
  ]
}